---
title: "Organise data"
author: "Sven Tobias-HÃ¼nefeldt"
date: "2023-11-28"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#Load packages
```{r}
library(RCurl)
library(tidyverse)
library(R.utils)
library(phyloseq)
library(ggplot2)
library(plotly)
library(forcats)
library(vegan)
library(dplyr)
library(ggpubr)
```

#Make environment and functions
```{r Set up environment}
#Set seed to make results more reproducible
set.seed(2)
#Set language to english
Sys.setenv(LANG = "en")
#Set up working directory
setwd("F:/Functional_R_analysis/")
#Define theme for plotting
My_Theme = theme_bw()+
  theme(axis.title.x = element_text(size=18),
       # theme_grey(base_size = 22),
        axis.text.x = element_text(angle=0, colour = "black", vjust=1, hjust = 0.5, size=18), 
        axis.text.y = element_text(colour = "black", size=18),
        axis.title.y = element_text(size=18),
        plot.title = element_text(size = 18, hjust = 0.5),
        legend.title =element_text(size = 18),
        legend.text = element_text(size = 16),
        legend.position="right",
        legend.key.size = unit(1, "cm"),
        strip.text.x = element_text(size=18, face="bold"),
        strip.text.y = element_text(size=18, face="bold"),
        #panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        strip.background = element_rect(colour="black"),
       text = element_text(family = "sans"))

Poster_Theme = theme_bw()+
  theme(axis.title.x = element_text(size=28),
       # theme_grey(base_size = 22),
        axis.text.x = element_text(angle=0, colour = "black", vjust=1, hjust = 0.5, size=24), 
        axis.text.y = element_text(colour = "black", size=24),
        axis.title.y = element_text(size=28),
        plot.title = element_text(size = 24, hjust = 0.5),
        legend.title =element_text(size = 28),
        legend.text = element_text(size = 24),
        legend.position="right",
        legend.key.size = unit(1, "cm"),
        strip.text.x = element_text(size=24, face="bold"),
        strip.text.y = element_text(size=24, face="bold"),
        #panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        strip.background = element_rect(colour="black"),
       text = element_text(family = "sans"))

#Station specific colours for consistency
Station_colour_list <- c("Muelenberger Loch" =  "black",
                "Twielenfleth" = "red",
                #"" = "violet",
                #"" = "violetred4", 
                "Schwarztonnensand" = "navy",
                #"" = "",
                #"" = "thistle3", 
                "Brunbuettel" = "forestgreen",
                #"" = "darkred",
                "Medemgrund" = "green",
                #"" = "orange",
                #"" = "turquoise",
                #"" = "cornflowerblue",
                #"" = "beige",
                "Station6" = "magenta")

Date_colour_list <- c("Jul 21" =  "forestgreen",
                "May 21" = "red",
                #"" = "violet",
                #"" = "violetred4", 
                "Nov 22" = "navy",
                #"" = "",
                #"" = "thistle3", 
                "Feb 22" = "black",
                #"" = "darkred",
                #"" = "turquoise",
                #"" = "cornflowerblue",
                #"" = "beige",
                #"" = "orange",
                "May 22" = "green",
                "Jun 22" = "magenta"
                )

#Colourblind friendly pallete
cbbPalette <- c("black",
                "red",
                "navy", 
                "green", 
                "magenta",
                "forestgreen")
#If more options are needed
expanded_cbbPalette <- c("#000000", #Black
                "#E69F00", #Orange
                "#56B4E9", #Blue (light)
                "#009E73", #Sea green
                "#CC79A7", #Magenta (light)
                "#F0E442", #Yellow
                "#0072B2", #Blue
                "#D55E00", #Burnt orange
                "dodgerblue4",
                "rosybrown",
                "floralwhite",
                "lightgoldenrod4",
                "cornsilk3",
                "coral4",
                "gray38",
                "turquoise4",
                "springgreen3",
                "slateblue3",
                "forestgreen",
                "lightgreen") 

#Shapes for ggplot2
Shape_list = c(0, #Hollow square
               15, #Filled square
               1, #Hollow circle
               16, #Filled circle
               2, #Hollow triangle
               17, #Filled triangle
               5, #Hollow diamond
               23) #Filled diamond
                
#Set up position dodge for ggplot2
pd = position_dodge(0.15)

ConditionColourList <- c("AlcianBlue" = "black",
                         "CBBG" = "red")

#Colourblind friendly pallete
FractionColourList <- c("Suspended" = "grey50",
                        "Sinking" = "black")
FractionSecondaryColourList <- c(#"Suspended" = "black",
                                 #"Sinking" = "red",
                                 #"navy",
                                 "Suspended" = "grey", 
                                 "Sinking" = "grey40")


Group_colours = c("Dissolved Organic" = "#000000", #Black
                                           "Dissolved Inorganic" =  "#E69F00", #Orange
                                           "Free-Living Organisms"= "#56B4E9", #Blue (light),
                                           "Suspended PAO" = "#009E73", #Sea green,
                                           "Sinking PAO" = "#CC79A7", #Magenta (light),
                                           "Suspended Organic Particles" = "#F0E442", #Yellow,
                                           "Suspended Inorganic Particles" = "#0072B2", #Blue,
                                           "Sinking Organic Particles" = "#D55E00", #Burnt orange,
                                           "Sinking Inorganic Particles" = "dodgerblue4")

```
```{r Hour:Minute to Hour function}
i="1"
#hours_minutes = Results.df$TimeDiff
hours_minutes_to_hours <- function(hours_minutes) {
  #print(length(hours_minutes))
  
  output =list()
  
  for (i in 1:length(hours_minutes)) {
    if(grepl(pattern =":",  x = hours_minutes[i]) == T) {
    
      splitNumbers = strsplit(hours_minutes[i], split = ":")
      minutes = as.numeric(splitNumbers[[1]][2])/60
      hours = as.numeric(splitNumbers[[1]][1])
    
      tmp_output = as.numeric(hours + minutes)
    
      output = c(output, tmp_output)
    
    }
    else if (grepl(pattern =".",  x = hours_minutes[i]) == T) {
      
      tmp_output = as.numeric(hours_minutes[i])
    
      output = c(output, tmp_output)
    
    }
  else {
    print("Sample did not match format, hours and minutes must be separated by :")
  }  
  }
  
  return(output)
}

```
```{r String cutting function}

substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

```
#Import dataframes and make analysis data frames
```{r}

path= "F:/Functional_R_analysis"
#path <- getwd()

h <- curl::new_handle()
curl::handle_setopt(
  handle = h,
  httpauth = 1,
  userpwd = "BICEST:vJGDYLs7"
)

#url_geneabund <- "https://sunagawalab.ethz.ch/share/BICEST/GROS22/MAGs/gene_catalog/GROS22__insertcounts.lengthnorm.profile.cellab.profile.gz" 
#curl::curl_download(url_geneabund,paste(path,"/BICEST_cellabund.tsv.gz", sep=""), handle =  h)
#gunzip(paste(path,"/BICEST_cellabund.tsv.gz", sep=""))
geneabund <- read.csv(paste0(path,"/BICEST_cellabund.tsv"), sep="\t", skip=1, header = TRUE, row.names = 1)

#url_annot <- "https://sunagawalab.ethz.ch/share/BICEST/GROS22/MAGs/gene_catalog/GROS22.kegg-annotations.tsv.gz" 
#curl::curl_download(url_annot,paste(path,"/GROS22.kegg-annotations.tsv.gz", sep=""), handle =  h)
#gunzip(paste(path,"/GROS22.kegg-annotations.tsv.gz", sep=""))
annot <- read.csv(paste0(path,"/GROS22.kegg-annotations.tsv"), sep="\t", skip=0, header = TRUE, row.names = 1, quote="")

#url_clstr <- "https://sunagawalab.ethz.ch/share/BICEST/GROS22/MAGs/gene_catalog/GROS22.clstr.gz" 
#curl::curl_download(url_clstr,paste(path,"/GROS22.clstr.gz", sep=""), handle =  h)
#gunzip(paste(path,"/GROS22.clstr.gz", sep=""))
clstr <- read.csv(paste0(path,"/GROS22.clstr"), sep="\t", skip=0, header = FALSE)
colnames(clstr) <- c("gene_cluster", "mem", "gene", "length", "pos")


#url_taxa <- "https://sunagawalab.ethz.ch/share/BICEST/GROS22/MAGs/metadata/GROS22-2.prok.genomes_2_gtdb.tsv.gz"
#curl::curl_download(url_taxa,paste(path,"/GROS22-2.gtdb.gz", sep=""), handle =  h)
#gunzip(paste(path,"/GROS22-2.gtdb.gz", sep=""))

#url_taxa2 <- "https://sunagawalab.ethz.ch/share/BICEST/GROS22/MAGs/metadata/GROS22-1.prok.genomes_2_gtdb.tsv.gz"
#curl::curl_download(url_taxa2,paste(path,"/GROS22-1.gtdb.gz", sep=""), handle =  h)
#gunzip(paste(path,"/GROS22-1.gtdb.gz", sep=""))

taxa <- read.csv(paste0(path,"/GROS22-2.gtdb"), sep="\t", skip=0, header = TRUE)
taxa <- rbind(taxa,read.csv(paste0(path,"/GROS22-2.gtdb"), sep="\t", header = TRUE))
taxa <- taxa %>% 
  separate(GTDBTK_TAXONOMY, c("domain", "phylum", "class", "order", "family",  "genus", "species"), ";")


#####
geneabund_2 <- geneabund %>% 
  rownames_to_column("gene_cluster") # remove rownames and rename to gene_cluster

geneabund_KO <- annot %>% 
  rownames_to_column("gene_cluster") %>% # remove rownames and rename to gene_cluster
  select(gene_cluster, KO) %>% # select only the columns titled gene_sluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values


geneabund_ko_otu_table <-
  geneabund_KO %>% 
  column_to_rownames("KO") %>% # remove rownames and rename to gene_cluster
  otu_table(., taxa_are_rows = TRUE) #make as a otu_table to generate a phyloseq object

sample_names(geneabund_ko_otu_table)
taxa_names(geneabund_ko_otu_table)

gene_abund_ko_annots <- annot %>%
 rownames_to_column("gene_cluster") %>% # remove rownames and rename to gene_cluster
  select(-"gene_cluster") %>% # remove the gene_cluster column
  distinct(., .keep_all = TRUE ) %>% # keep only unique KO id's
  rownames_to_column("gene_cluster") %>% # remove rownames and rename to gene_cluster
  select(KO, everything()) %>%  # reorders - so KO is in front
  column_to_rownames("KO") %>% # remove column named KO and make it the rowname
  as.matrix() %>% # convert to a matrix
  tax_table() # convert to a taxa table object for phyloseq object creation


taxa_names(gene_abund_ko_annots)


#####
metadata <- read.csv("SAMEAID_SampleID_simplified.csv", header=TRUE, sep=";") %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAT.genecount.profile")) %>% # Make metatranscriptome metadata file 
  mutate(data_type="METAT") # Add column that says METAT


metadata2 <- read.csv("SAMEAID_SampleID_simplified.csv", header=TRUE, sep=";") %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAG.genecount.profile"))%>% # Make metagenome metadata file
  mutate(data_type="METAG") # Add column that says METAG

metadata <- rbind(metadata, metadata2) # Combine metadata files

rm(metadata2)

metadata$Station = gsub("Meedem Grund", "Medemgrund", metadata$Station) 

#Clean up data  
metadata$station_km = metadata$Stromkilometer
metadata$station_km = gsub(608.165, 608, metadata$station_km)
metadata$station_km = gsub(613, 713, metadata$station_km)
metadata$station_km = gsub(632.88, 633, metadata$station_km)
metadata$station_km = gsub(632.884, 633, metadata$station_km)
metadata$station_km = gsub(633.022, 633, metadata$station_km)
metadata$station_km = gsub(6334, 633, metadata$station_km)
metadata$station_km = gsub(651.32, 651, metadata$station_km)
metadata$station_km = gsub(651.323, 651, metadata$station_km)
metadata$station_km = gsub(651.955, 651, metadata$station_km)
metadata$station_km = gsub(6513, 651, metadata$station_km)
metadata$station_km = gsub(665.41, 665, metadata$station_km)
metadata$station_km = gsub(665.414, 665, metadata$station_km)
metadata$station_km = gsub(665.546, 665, metadata$station_km)
metadata$station_km = gsub(6654, 665, metadata$station_km)
metadata$station_km = gsub(691.997, 692, metadata$station_km)
metadata$station_km = gsub(692.010, 692, metadata$station_km)
metadata$station_km = gsub(692.01, 692, metadata$station_km)
metadata$station_km = gsub(694, 692, metadata$station_km)
metadata$station_km = gsub(711.515, 713, metadata$station_km)
metadata$station_km = gsub(712, 713, metadata$station_km)
metadata$station_km = gsub(714.975, 713, metadata$station_km)
metadata$station_km = gsub(714.98, 713, metadata$station_km)
sort(unique(metadata$station_km))

metadata = subset(metadata, station_km > 630)
metadata$station_km = as.numeric(metadata$station_km)
metadata$Stromkilometer = NULL

metadata$Sample_date = gsub("-", " ", metadata$Sample_date)
metadata$Sample_date = gsub("Mai", "May", metadata$Sample_date)


genecat_ps <- phyloseq(geneabund_ko_otu_table, gene_abund_ko_annots, metadata)  # Combine into a single phyloseq object


gene_cluster_taxa <- clstr %>% 
  select("gene_cluster", "gene") %>%  # retrieve the gene cluster and gene column
  mutate(genome=str_replace(gene, "-scaf.*", "")) # remove the scaffold information to only keep the MAG ID the gene cluster is associated with.

```
#amoA
```{r}

# make taxonomy table
amoa_tbl <- annot %>% 
  rownames_to_column("gene_cluster") %>%  # remove rownames and rename to gene_cluster
  filter(KO == "K10944") %>% # only extract the amoA gene
  select(gene_cluster) %>% # only get the gene cluster id
  left_join(gene_cluster_taxa) %>%  # Combine gene_cluster information with the genome information
  left_join(taxa, by = c("genome" = "GENOME"), relationship= "many-to-many") %>% # add taxonomic information based on genome name
  select(gene_cluster, phylum, class, order, family, genus) %>% # extract only the gene_cluster and taxonomic information
  distinct() %>% # make sure each line is unique
  filter(phylum != "NA") # remove any rows that are not associated with a specific phyla

# make abundance table, subset by taxonomy 
amoA_otu_tbl <- annot %>% 
  rownames_to_column("gene_cluster") %>%  # remove rownames and rename to gene_cluster
  filter(KO == "K10944") %>% # only extract the amoA gene
  select(gene_cluster) %>%  # only get the gene cluster id
  left_join(gene_cluster_taxa) %>%  # Combine gene_cluster information with the genome information
  left_join(taxa, by = c("genome" = "GENOME"), relationship= "many-to-many") %>% # add taxonomic information based on genome name
  select(gene_cluster, phylum, class, order, family, genus) %>% # extract only the gene_cluster and taxonomic information
  distinct() %>% # make sure each line is unique
  filter(phylum != "NA") %>% # remove any rows that are not associated with a specific phyla
  filter(grepl("Nitro", genus)) %>% # Extract only genus that include "Nitro" in their name
  select(gene_cluster) %>% # extract gene_cluster information
  left_join(geneabund_2) # match these gene clusters with the abundance information

# make dataframe ready to plot by combining abundance with taxonomy information in a long format
amoA_otu_tbl_long <- amoA_otu_tbl %>%
  select(-length) %>% # remove length column
  pivot_longer(!gene_cluster, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(amoa_tbl, relationship = "many-to-many") %>%  # combine this with the taxonomic information we previously generated
  left_join(metadata)%>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') # remove sample from BunthausSpitze

#Reorder factors for better plotting
amoA_otu_tbl_long$Station = factor(amoA_otu_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoA_otu_tbl_long$Sample_date = factor(amoA_otu_tbl_long$Sample_date,
                                   levels = c("Mai 21",
                                              "Jul 21",
                                              "Feb 22",
                                              "Mai 22",
                                              "Jun 22",
                                              "Nov 22"))


amoA_otu_tbl_long$counts = amoA_otu_tbl_long$counts/1000
# Plot
ggplot(amoA_otu_tbl_long) + 
  geom_tile(aes(x=Station, y=genus, fill=log(counts))) +
  facet_wrap(~ data_type *Sample_date )+
  scale_fill_gradient2(low= "green", mid = "white", high = "red", na.value = "green")+
  ggtitle("amoA")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

tiff("amoA_genecatalogue.tiff", res = 100, units = "in", width = 10, height = 15)
last_plot()
dev.off()

```
This is still normalised to a per genome count based on the whole sample

#Carbon producing vs utilising
 - use this paper ?
 Genome content predicts the carbon catabolic preferences of heterotrophic bacteria - Otto Cordero

##Heat plot of CO2/CH4 utilising genes
```{r}
#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% BugHunt_GeneList.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% BugHunt_GeneList.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
  
amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% BugHunt_GeneList.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(BugHunt_GeneList.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') # remove sample from BunthausSpitze
  

dim(amoAClean_CO2CH4_tbl_long)

#CO2CH4_tbl_long %>% separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName"))

#Reorder factors for better plotting
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("Mai 21",
                                              "Jul 21",
                                              "Feb 22",
                                              "Mai 22",
                                              "Jun 22",
                                              "Nov 22"))

#Adjust values for accuracy and log transform
amoAClean_CO2CH4_tbl_long$counts = (amoAClean_CO2CH4_tbl_long$counts/1000)#+1
#CO2CH4_tbl_long$counts = CO2CH4_tbl_long$counts + 1

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)
```
##Significant fraction/date/Station differences
```{r}

####Prepare dataframe####


#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2) %>%
  distinct()

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")


####Run tests####

amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")
#Remove all but the relevant object to run anova models
#library(gdata)
#keep(amoAClean_CO2CH4_tbl_long)
#keep(amoAClean_CO2CH4_tbl_long, sure = TRUE, all = TRUE)

#CO2CH4_tbl_long %>% separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName"))

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("Mai 21",
                                              "Jul 21",
                                              "Feb 22",
                                              "Mai 22",
                                              "Jun 22",
                                              "Nov 22"))

#Adjust values for accuracy and log transform
#amoAClean_CO2CH4_tbl_long$counts = (amoAClean_CO2CH4_tbl_long$counts/1000)#+1
#CO2CH4_tbl_long$counts = CO2CH4_tbl_long$counts + 1

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)

#BugHunt.df = amoAClean_CO2CH4_tbl_long[c(1:200000),]


uniquegenes.ls = unique(amoAClean_CO2CH4_tbl_long$SName)
uniquegenes.ls


#Make results table
Results.df = data.frame(Gene = "DELETEME",
                        Sample_type.eta2 = 9999,
                        Sample_type.p = 9999,
                        Station.eta2 = 9999,
                        Station.p = 9999,
                        Sample_date.eta2 = 9999,
                        Sample_date.p = 9999,
                        "Sample_type:Station.eta2" = 9999,
                        "Sample_type:Station.p" = 9999,
                        "Sample_type:Sample_date.eta2" = 9999,
                        "Sample_type:Sample_date.p" = 9999,
                        "Station:Sample_date.eta2" = 9999,
                        "Station:Sample_date.p" = 9999,
                        "Sample_type:Station:Sample_date.eta2" = 9999,
                        "Sample_type:Station:Sample_date.p" = 9999,
                        "TukeyGroupsFraction<0.05" = paste(tmp_G.ptuk, collapse = ','),
                        data_type = "DELETEME")

i="ME2, sfcA, maeA"
#Get significant differences over space
for (i in uniquegenes.ls) {
  print(paste0("Working on ", i))
  
  tmp_G.sbst = subset(amoAClean_CO2CH4_tbl_long, SName == i & data_type == "Metagenomes")
  #tmp_G.sbst
  
  tmp_T.sbst = subset(amoAClean_CO2CH4_tbl_long, SName == i & data_type =="Transcriptomes")
  #tmp_T.sbst
  
  tmp_G.aov = aov(counts ~ Sample_type * Station * Sample_date, data = tmp_G.sbst)
  tmp_T.aov = aov(counts ~ Sample_type * Station * Sample_date, data = tmp_T.sbst)
  
  #summary(tmp_G.aov)
  #summary(tmp_T.aov)
  
  tmp_G.eta2 = lsr::etaSquared(tmp_G.aov)
  tmp_T.eta2 = lsr::etaSquared(tmp_T.aov)
  
  tmp_G.tuk = TukeyHSD(tmp_G.aov)
  tmp_T.tuk = TukeyHSD(tmp_T.aov)
  
  colnames(tmp_G.tuk$Sample_type) = gsub(" ", "_", colnames(tmp_G.tuk$Sample_type))
  tmp_G.ptuk = tmp_G.tuk$Sample_type %>%
    as.data.frame() %>%
    filter(p_adj < 0.05) %>%
    rownames()
  colnames(tmp_T.tuk$Sample_type) = gsub(" ", "_", colnames(tmp_T.tuk$Sample_type))
  tmp_T.ptuk = tmp_T.tuk$Sample_type %>%
    as.data.frame() %>%
    filter(p_adj < 0.05) %>%
    rownames()
  
  #tmp_G.ptuk
  #tmp_T.ptuk
  
  
  tmp_G.res = data.frame(Gene = paste0(i),
                         Sample_type.eta2 = tmp_G.eta2[1,1],
                         Sample_type.p = summary(tmp_G.aov)[[1]][["Pr(>F)"]][1],
                         Station.eta2 = tmp_G.eta2[2,1],
                         Station.p = summary(tmp_G.aov)[[1]][["Pr(>F)"]][2],
                         Sample_date.eta2 = tmp_G.eta2[3,1],
                         Sample_date.p = summary(tmp_G.aov)[[1]][["Pr(>F)"]][3],
                         "Sample_type:Station.eta2" = tmp_G.eta2[4,1],
                         "Sample_type:Station.p" = summary(tmp_G.aov)[[1]][["Pr(>F)"]][4],
                         "Sample_type:Sample_date.eta2" = tmp_G.eta2[5,1],
                         "Sample_type:Sample_date.p" = summary(tmp_G.aov)[[1]][["Pr(>F)"]][5],
                         "Station:Sample_date.eta2" = tmp_G.eta2[6,1],
                         "Station:Sample_date.p" = summary(tmp_G.aov)[[1]][["Pr(>F)"]][6],
                         "Sample_type:Station:Sample_date.eta2" = tmp_G.eta2[7,1],
                         "Sample_type:Station:Sample_date.p" = summary(tmp_G.aov)[[1]][["Pr(>F)"]][7],
                         "TukeyGroupsFraction<0.05" = paste(tmp_G.ptuk, collapse = ','),
                         data_type = "Metagenomes")
  tmp_T.res = data.frame(Gene = paste0(i),
                         Sample_type.eta2 = tmp_T.eta2[1,1],
                         Sample_type.p = summary(tmp_T.aov)[[1]][["Pr(>F)"]][1],
                         Station.eta2 = tmp_T.eta2[2,1],
                         Station.p = summary(tmp_T.aov)[[1]][["Pr(>F)"]][2],
                         Sample_date.eta2 = tmp_T.eta2[3,1],
                         Sample_date.p = summary(tmp_T.aov)[[1]][["Pr(>F)"]][3],
                         "Sample_type:Station.eta2" = tmp_T.eta2[4,1],
                         "Sample_type:Station.p" = summary(tmp_T.aov)[[1]][["Pr(>F)"]][4],
                         "Sample_type:Sample_date.eta2" = tmp_T.eta2[5,1],
                         "Sample_type:Sample_date.p" = summary(tmp_T.aov)[[1]][["Pr(>F)"]][5],
                         "Station:Sample_date.eta2" = tmp_T.eta2[6,1],
                         "Station:Sample_date.p" = summary(tmp_T.aov)[[1]][["Pr(>F)"]][6],
                         "Sample_type:Station:Sample_date.eta2" = tmp_T.eta2[7,1],
                         "Sample_type:Station:Sample_date.p" = summary(tmp_T.aov)[[1]][["Pr(>F)"]][7],
                         "TukeyGroupsFraction<0.05" = paste(tmp_T.ptuk, collapse = ','),
                         data_type = "Transcriptomes")
  
  
  Results.df = rbind(Results.df, tmp_G.res, tmp_T.res)
}
Results.df = Results.df[-1,]
Results.df


Results.df$TukeyGroupsFraction.0.05 = gsub("_fraction", "", Results.df$TukeyGroupsFraction.0.05)
Results.df$TukeyGroupsFraction.0.05 = gsub("_living", "", Results.df$TukeyGroupsFraction.0.05)


Results_Fraction.df = subset(Results.df, Sample_type.p < 0.05)
Results_Station.df = subset(Results.df, Station.p < 0.05)
Results_Date.df = subset(Results.df, Sample_date.p < 0.05)


####Combine and export####
#Results.df = rbind(Results_Utilising.df, Results_Producing.df, Results_Both.df, Results_Unknown.df)

write.csv(Results_Fraction.df, "GeneSignificant_Fraction.csv")
write.csv(Results_Station.df, "GeneSignificant_Station.csv")
write.csv(Results_Date.df, "GeneSignificant_Date.csv")

```

##Significant changes associated with salinity

```{r}

####Create dataframe####
#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2)

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

####Run test####

amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")
#Remove all but the relevant object to run anova models
#library(gdata)
#keep(amoAClean_CO2CH4_tbl_long)
#keep(amoAClean_CO2CH4_tbl_long, sure = TRUE, all = TRUE)

#CO2CH4_tbl_long %>% separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName"))

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("Mai 21",
                                              "Jul 21",
                                              "Feb 22",
                                              "Mai 22",
                                              "Jun 22",
                                              "Nov 22"))

#Adjust values for accuracy and log transform
#amoAClean_CO2CH4_tbl_long$counts = (amoAClean_CO2CH4_tbl_long$counts/1000)#+1
#CO2CH4_tbl_long$counts = CO2CH4_tbl_long$counts + 1

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)

#BugHunt.df = amoAClean_CO2CH4_tbl_long[c(1:200000),]

Complete_Metadata.df = read.csv("SAMEAID_SampleID.csv", sep = ";", header = T)[,c(1,10:15)]

amoAClean_CO2CH4_tbl_long = amoAClean_CO2CH4_tbl_long %>%
  select(-X) %>%
  left_join(Complete_Metadata.df, by = c("BioSample" = "AccessionNumber_TBDSven")) %>%
  mutate_at(c('counts'), as.numeric) %>%
  distinct()
dim(amoAClean_CO2CH4_tbl_long)
amoAClean_CO2CH4_tbl_long$Salinity_TBDHereon = as.numeric(gsub(",", ".", gsub("\\.", "", amoAClean_CO2CH4_tbl_long$Salinity_TBDHereon)))


uniquegenes.ls = unique(amoAClean_CO2CH4_tbl_long$SName)
uniquegenes.ls


#Make results table
Results_Sal.df = data.frame(Gene = "DELETEME",
                        Salinity.rho = 9999,
                        Salinity.p = 9999,
                        data_type = "DELETEME")

i="ME2, sfcA, maeA"
#Get significant differences over space
for (i in uniquegenes.ls) {
  print(paste0("Working on ", i))
  
  tmp_G.sbst = subset(amoAClean_CO2CH4_tbl_long, SName == i & data_type == "Metagenomes")
  tmp_G.sbst
  
  tmp_T.sbst = subset(amoAClean_CO2CH4_tbl_long, SName == i & data_type =="Transcriptomes")
  tmp_T.sbst
  
  tryCatch({
    tmp_G.spear = cor.test(tmp_G.sbst$counts, tmp_G.sbst$Salinity_TBDHereon, method = "s")
    tmp_T.spear = cor.test(tmp_T.sbst$counts, tmp_T.sbst$Salinity_TBDHereon, method = "s")
    tmp_G.res = data.frame(Gene = paste0(i),
                         Salinity.rho = tmp_G.spear$estimate,
                         Salinity.p = tmp_G.spear$p.value,
                         data_type = "Metagenomes")
    tmp_T.res = data.frame(Gene = paste0(i),
                         Salinity.rho = tmp_T.spear$estimate,
                         Salinity.p = tmp_T.spear$p.value,
                         data_type = "Transcriptomes")
  
  
   Results_Sal.df = rbind(Results_Sal.df, tmp_G.res, tmp_T.res)}, 
    
   error=function(e){cat("ERROR :", conditionMessage(e), "\n")})
  
  
  
  
}
Results_Sal.df = Results_Sal.df[-1,]
Results_Sal.df
dim(Results_Sal.df)

Results_Sal.df = subset(Results_Sal.df, Salinity.p < 0.05)
dim(Results_Sal.df)

####Combine and export####
#Results.df = rbind(Results_Utilising.df, Results_Producing.df, Results_Both.df, Results_Unknown.df)

Results_Sal.df$TukeyGroupsFraction.0.05 = gsub("_fraction", "", Results_Sal.df$TukeyGroupsFraction.0.05)
Results_Sal.df$TukeyGroupsFraction.0.05 = gsub("_living", "", Results_Sal.df$TukeyGroupsFraction.0.05)

write.csv(Results_Sal.df, "GeneSignificant_Salinity.csv")


```
##Plots
###Fractions
```{r}

#Read in dataframe
SignGenes_Fraction.df = read.csv("GeneSignificant_Fraction.csv", header = T, sep = ";")

SignGenes_Fraction.df = subset(SignGenes_Fraction.df, Sample_type.eta2 > 0.3)
dim(SignGenes_Fraction.df)
# Plot preferences
amoAClean_CO2CH4Profiles.plt = ggplot(SignGenes_Fraction.df) + 
  geom_tile(aes(x=data_type, y=fct_reorder(.f = Gene, .x = Sample_type.eta2, .fun = mean, .na_rm = T, .desc = F), fill=Sample_type.eta2)) +
  #facet_wrap(Direction ~ data_type * Sample_date, ncol = 9)+
  scale_fill_gradient2(expression(paste("eta"^{2})), low= "magenta", mid = "white", high = "black", na.value = "red")+
  ggtitle("CO2/CH4")+
  ylab("Significantly changing \nCO2/CH4 Genes")+
  xlab("Sequencing type")+
  My_Theme+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# visualise
amoAClean_CO2CH4Profiles.plt 

#Save plots
#pdf("Figures/CO2CH4_Unknown Reversibility.pdf", width = 12, height = 12)
#CO2CH4Profiles.plt
#dev.off()

png("Figures/CO2CH4_Signfraction0.3.png", width = 10, height = 12, units = "in", res = 120)
amoAClean_CO2CH4Profiles.plt
dev.off()




```

###Salinity

```{r}

#Read in dataframe
SignGenes_Salinity.df = read.csv("GeneSignificant_Salinity.csv", header = T, sep = ";")

SignGenes_Salinity.df = subset(SignGenes_Salinity.df, Sample_type.eta2 > 0.3)
dim(SignGenes_Salinity.df)
# Plot preferences
amoAClean_CO2CH4Profiles.plt = ggplot(SignGenes_Salinity.df) + 
  geom_tile(aes(x=data_type, y=fct_reorder(.f = Gene, .x = Sample_type.eta2, .fun = mean, .na_rm = T, .desc = F), fill=Sample_type.eta2)) +
  #facet_wrap(Direction ~ data_type * Sample_date, ncol = 9)+
  scale_fill_gradient2(expression(paste("eta"^{2})), low= "magenta", mid = "white", high = "black", na.value = "red")+
  ggtitle("CO2/CH4")+
  ylab("Significantly changing \nCO2/CH4 Genes")+
  xlab("Sequencing type")+
  My_Theme+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# visualise
amoAClean_CO2CH4Profiles.plt 

#Save plots
#pdf("Figures/CO2CH4_Unknown Reversibility.pdf", width = 12, height = 12)
#CO2CH4Profiles.plt
#dev.off()

png("Figures/CO2CH4_SignSalinity0.3.png", width = 30, height = 45, units = "in", res = 120)
amoAClean_CO2CH4Profiles.plt
dev.off()




```




#NMDS
##Metagenome
```{r}
#Add relevant rownames for making the phyloseq object
rownames(metadata) = metadata$sampleid
#Change names for consistency
metadata$Sample_date = gsub("Mai", "May", metadata$Sample_date)
#Reorder station km 
metadata$Station = factor(metadata$Station,
                          levels = c("Medemgrund",
                                     "Brunsbuettel",
                                     "Schwarztonnensand",
                                     "Twielenfleth",
                                     "Muehlenberger Loch",
                                     "Kollmar"))
#Reorder Dates
metadata$Sample_date = factor(metadata$Sample_date,
                          levels = c("May 21",
                                     "Jul 21",
                                     "Feb 22",
                                     "May 22",
                                     "Jun 22",
                                     "Nov 22"))


#Make phyloseq object
genecat_ps <- phyloseq(geneabund_ko_otu_table, gene_abund_ko_annots, sample_data(metadata))

#Subset to samples we want
genecat_pgene_abund_ko_annotsgenecat_ps = subset_samples(genecat_ps, data_type=="METAG")


#Clean up samples for the relevant ones
genecat_LH_ps = subset_samples(genecat_pgene_abund_ko_annotsgenecat_ps, Sample_type!="Free_living")
genecat_LH.ps = subset_samples(genecat_LH_ps, station_km > 630 & Sample_date!="Nov 21" )
genecat_LH.ps = subset_samples(genecat_LH.ps, station_km!=713 | Sample_date!="Jun 22")

#Calculate ordination matrix based on Bray-Curtis dissimilarity
Gene.ord <- ordinate(genecat_LH.ps, method = "NMDS", distance = "bray")
#Ensure data is good fit
stressplot(Gene.ord) #Good stress plot
Gene.ord #Good level of stress - 0.1199889

####Make plots ####
#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Sample_date",
                             shape = "Station"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Sample_date, color = Sample_date), lwd = 1)+
  scale_color_manual("Sample Date", values = Date_colour_list)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  geom_point(aes(size = 2, color = Sample_date), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0.2, y=0.2, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("Figures/Gene_METAG_NMDS.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - Sample_date
Sample_date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, Sample_date_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#May 22 and Jun 22 and Jun 22 and Nov 22 match. All others are unique.

#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Station", 
                             shape = "Sample_date"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Station, color = Station), lwd = 1)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  scale_color_manual("Sample Date", values = cbbPalette)+
  geom_point(aes(size = 2, color = Station), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0.2, y=0.2, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("Figures/Gene_METAG_NMDS2.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - Stations
Station_group = get_variable(genecat_LH.ps@sam_data, "Station")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, Station_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#Patterns exist in the station differences - all except last estuary stations are similar to previous and subsequent stations

#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Sample_type", 
                             shape = "Station"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Sample_type, color = Sample_type), lwd = 1)+
  scale_color_manual("Particle Fraction", values = cbbPalette)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  geom_point(aes(size = 2, color = Sample_type), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0.15, y=0.2, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("Figures/Gene_METAG_NMDS3.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - fractions
fraction_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, fraction_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#L and H are not significantly different



####Anosim, adonis, and permanova date tests####

#anosim
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
date_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), date_group, distance = "bray")
date_ano$signif #0.001
date_ano$statistic #0.4385365
#Significant differences between group means - based on date

#Adonis
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
date_ado = adonis2(Gene.dist ~ date_group)
date_ado
#Significant differences between group means and distribution - based on date
#R2 = 0.39111 
#p = 0.001

#Pairwise PERMANOVA
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
date.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, date_group, nperm = 999, p.method = "bonferroni")
date.stat
#May 22 and Jun 22 and Jun 22 and Nov 22 match. All others are unique.


####Anosim and adonis station_km tests####

#anosim
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
station_km_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), station_km_group, distance = "bray")
station_km_ano$signif #0.001
station_km_ano$statistic #0.2464986
#Significant differences between group means - based on station_km

#Adonis
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km_ado = adonis2(Gene.dist ~ station_km_group)
station_km_ado
#Significant differences between group means and distribution - based on station_km
#R2 = 0.17356
#p = 0.001

#Pairwise PERMANOVA
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, station_km_group, nperm = 999, p.method = "bonferroni")
station_km.stat
#Patterns exist in the station differences - all except last estuary stations are similar to previous and subsequent stations


####Anosim and adonis fraction tests####

#anosim
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
station_km_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), station_km_group, distance = "bray")
station_km_ano$signif #0.545
station_km_ano$statistic #-0.003540322
#Nosignificant differences between group means

#Adonis
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km_ado = adonis2(Gene.dist ~ station_km_group)
station_km_ado
#No significant differences between group means and distribution - based on fraction
#R2 = 0.00696 
#p = 0.586

#Pairwise PERMANOVA
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, station_km_group, nperm = 999, p.method = "bonferroni")
station_km.stat
#No significant fraction difference


```
##Metatranscriptome
```{r}

####Metadata####
metadata <- read.csv("F:/Functional_R_analysis/SAMEAID_SampleID_simplified.csv", header=TRUE, sep=";") %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAT.genecount.profile")) %>% # Make metatranscriptome metadata file 
  mutate(data_type="METAT") # Add column that says METAT


metadata2 <- read.csv("F:/Functional_R_analysis/SAMEAID_SampleID_simplified.csv", header=TRUE, sep=";") %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAG.genecount.profile"))%>% # Make metagenome metadata file
  mutate(data_type="METAG") # Add column that says METAG

metadata <- rbind(metadata, metadata2) # Combine metadata files

rm(metadata2)

metadata$Station = gsub("Meedem Grund", "Medemgrund", metadata$Station) 

#Clean up data  
metadata$station_km = metadata$Stromkilometer
metadata$station_km = gsub(608.165, 608, metadata$station_km)
metadata$station_km = gsub(613, 713, metadata$station_km)
metadata$station_km = gsub(632.88, 633, metadata$station_km)
metadata$station_km = gsub(632.884, 633, metadata$station_km)
metadata$station_km = gsub(633.022, 633, metadata$station_km)
metadata$station_km = gsub(6334, 633, metadata$station_km)
metadata$station_km = gsub(651.32, 651, metadata$station_km)
metadata$station_km = gsub(651.323, 651, metadata$station_km)
metadata$station_km = gsub(651.955, 651, metadata$station_km)
metadata$station_km = gsub(6513, 651, metadata$station_km)
metadata$station_km = gsub(665.41, 665, metadata$station_km)
metadata$station_km = gsub(665.414, 665, metadata$station_km)
metadata$station_km = gsub(665.546, 665, metadata$station_km)
metadata$station_km = gsub(6654, 665, metadata$station_km)
metadata$station_km = gsub(691.997, 692, metadata$station_km)
metadata$station_km = gsub(692.010, 692, metadata$station_km)
metadata$station_km = gsub(692.01, 692, metadata$station_km)
metadata$station_km = gsub(694, 692, metadata$station_km)
metadata$station_km = gsub(711.515, 713, metadata$station_km)
metadata$station_km = gsub(712, 713, metadata$station_km)
metadata$station_km = gsub(714.975, 713, metadata$station_km)
metadata$station_km = gsub(714.98, 713, metadata$station_km)
sort(unique(metadata$station_km))

metadata = subset(metadata, station_km > 630)
metadata$station_km = as.numeric(metadata$station_km)
metadata$Stromkilometer = NULL

metadata$Sample_date = gsub("-", " ", metadata$Sample_date)
metadata$Sample_date = gsub("Mai", "May", metadata$Sample_date)

#Add relevant rownames for making the phyloseq object
rownames(metadata) = metadata$sampleid
#Change names for consistency
metadata$Sample_date = gsub("Mai", "May", metadata$Sample_date)
#Reorder station km 
metadata$Station = factor(metadata$Station,
                          levels = c("Medemgrund",
                                     "Brunsbuettel",
                                     "Schwarztonnensand",
                                     "Twielenfleth",
                                     "Muehlenberger Loch",
                                     "Kollmar"))
#Reorder Dates
metadata$Sample_date = factor(metadata$Sample_date,
                          levels = c("May 21",
                                     "Jul 21",
                                     "Feb 22",
                                     "May 22",
                                     "Jun 22",
                                     "Nov 22"))


####OTU table ####


#Import gene list
CarbonGenelist.df = read.csv("F:/Functional_R_analysis/CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2) %>%
  distinct()

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = gsub("-", " ",amoAClean_CO2CH4_tbl_long$Sample_date)

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("May 21",
                                              "Jul 21",
                                              "Nov 21",
                                              "Feb 22",
                                              "May 22",
                                              "Jun 22",
                                              "Nov 22"))

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)


#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
CO2CH4_TG.df <- spread(amoAClean_CO2CH4_tbl_long[ , ! names(amoAClean_CO2CH4_tbl_long) %in% c("sampleid", "abundance", "occurrence", "X")], #Remove columns that will cause problems
                       data_type, #Column that will contain new column names
                       counts) %>% # Value that will fill new columns
  mutate_at(c("Metagenomes", "Transcriptomes"), ~replace(., is.na(.), 0)) #Replace NAs with 0 in the new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_tbl_long)
dim(CO2CH4_TG.df)




#Calculate Transcripts per gene copy
CO2CH4_TG.df$TranscriptsPerGenome = CO2CH4_TG.df$Transcriptomes / CO2CH4_TG.df$Metagenomes

#Add more info - remove NA's and Infinite numbers
CO2CH4_TG.df = CO2CH4_TG.df %>%
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.na(.), 0)) %>% # This was ususally the case when 0/0
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.infinite(.), "9999999999")) #Replace Infin with #/0, as Metagenome likely incomplete

#Convert from wide to long format
CO2CH4_TG.lng = gather(CO2CH4_TG.df, key = data_type, value = counts, Metagenomes:TranscriptsPerGenome)
dim(CO2CH4_TG.lng)

#Remove data_types and columns that are no longer relevant 
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, data_type == "TranscriptsPerGenome") %>%
  select(counts, KO, Associatednumber)
dim(CO2CH4_TG.lng)


#Remove samples for which we have no metatranscriptomes
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, Associatednumber > 116)

#Remove samples for which sequencing failed
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, !(Associatednumber %in% c(127, 128, 140, 145, 146, 157, 169, 170,
                                                               119, 125, 126, 136, 137, 153, 154)))
#Make sure data is set up properly
CO2CH4_TG.lng$counts = as.numeric(as.character(CO2CH4_TG.lng$counts))
CO2CH4_TG.lng$Associatednumber = as.character(CO2CH4_TG.lng$Associatednumber)


Complete_Metadata.df = read.csv("F:/Functional_R_analysis/SAMEAID_SampleID.csv", sep = ";", header = T)[,c(1,3)]
typeof(Complete_Metadata.df$Associatednumber)
typeof(CO2CH4_TG.lng$Associatednumber)

Complete_Metadata.df$Associatednumber = as.character(Complete_Metadata.df$Associatednumber)

typeof(Complete_Metadata.df$Associatednumber)
typeof(CO2CH4_TG.lng$Associatednumber)

amoAClean_CO2CH4_wide_MT_mod = CO2CH4_TG.lng %>%
  #select(-X) %>%
  left_join(Complete_Metadata.df, by = c("Associatednumber" = "Associatednumber")) %>%
  mutate_at(c('counts'), as.numeric) %>%
  distinct()
dim(amoAClean_CO2CH4_wide_MT_mod)
amoAClean_CO2CH4_wide_MT_mod$Associatednumber = NULL

amoAClean_CO2CH4_wide_MT_mod$AccessionNumber_TBDSven = paste0("GROS22.2_", amoAClean_CO2CH4_wide_MT_mod$AccessionNumber_TBDSven, "_METAT.genecount.profile")

#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
amoAClean_CO2CH4_wide_MT_mod <- data.frame(spread(unique(amoAClean_CO2CH4_wide_MT_mod),
                       key = AccessionNumber_TBDSven, #Column that will contain new column names
                       value = counts)) # Value that will fill new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_wide_MT_mod)
View(amoAClean_CO2CH4_wide_MT_mod)




#Apply row names from gene KOs
rownames(amoAClean_CO2CH4_wide_MT_mod) = amoAClean_CO2CH4_wide_MT_mod$KO
amoAClean_CO2CH4_wide_MT_mod$KO = NULL


####Combine and plot ####

#Ensure only relevant rows of metadata make it into the phyloseq object
metadata = subset(metadata, sampleid %in% colnames(amoAClean_CO2CH4_wide_MT_mod))


#Make phyloseq object
genecat_MT_ps <- phyloseq(otu_table(amoAClean_CO2CH4_wide_MT_mod, taxa_are_rows = T), gene_abund_ko_annots, sample_data(metadata))

#Clean up samples for the relevant ones
genecat_LH_ps = subset_samples(genecat_MT_ps, Sample_type!="Free_living")
genecat_LH.ps = subset_samples(genecat_LH_ps, station_km > 630 & Sample_date!="Nov 21" )
genecat_LH.ps = subset_samples(genecat_LH.ps, station_km!=713 | Sample_date!="Jun 22")

#Calculate ordination matrix based on Bray-Curtis dissimilarity
Gene.ord <- ordinate(genecat_LH.ps, method = "NMDS", distance = "bray")
#Ensure data is good fit
stressplot(Gene.ord) #Decent stress plot - seems to have two distinct groups
Gene.ord #Good level of stress - 0.06460416

#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Sample_date",
                             shape = "Station"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Sample_date, color = Sample_date), lwd = 1)+
  scale_color_manual("Sample Date", values = Date_colour_list)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  geom_point(aes(size = 2, color = Sample_date), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0, y=0.48, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("F:/Functional_R_analysis/Figures/Gene_METAT_NMDS.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - Sample_date
Sample_date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, Sample_date_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#All are significantly different - however some marginally less so

#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Station", 
                             shape = "Sample_date"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Station, color = Station), lwd = 1)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  scale_color_manual("Sample Date", values = cbbPalette)+
  geom_point(aes(size = 2, color = Station), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0.2, y=0.2, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("F:/Functional_R_analysis/Figures/Gene_METAT_NMDS2.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - Stations
Station_group = get_variable(genecat_LH.ps@sam_data, "Station")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, Station_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#Two groups of samples, seems to be salinity based MG/BB/ST vs TW/ML 

#Design plot
NMDS_Gene.plt = plot_ordination(genecat_LH.ps, 
                             Gene.ord, 
                             color = "Sample_type", 
                             shape = "Station"
                             ) + 
  theme(legend.position = "right")+
  stat_ellipse(aes(group = Sample_type, color = Sample_type), lwd = 1)+
  scale_color_manual("Particle Fraction", values = cbbPalette)+
  scale_shape_manual("Elbe Station", values = Shape_list)+
  geom_point(aes(size = 2, color = Sample_type), show.legend = F)+
  #facet_grid(. ~ data_type)+
  annotate("text", x=0.15, y=0.2, size = 6, label= paste0("Stress = ", round(Gene.ord$stress, digits = 3)))+
  theme_bw()+
  My_Theme+
  guides(colour = guide_legend(order = 1),
         shape = guide_legend(order = 2, override.aes = list(size = 4)),
         size = "none",
         linewidth = "none")
#Inspect plot
#NMDS_Gene.plt$layers = NMDS_Gene.plt$layers[-1]
NMDS_Gene.plt

png("F:/Functional_R_analysis/Figures/Gene_METAT_NMDS3.png", width = 10, height = 7, units = "in", res = 120)
NMDS_Gene.plt
dev.off()

#Pairwise PERMANOVA - fractions
fraction_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
Depth_ado = RVAideMemoire::pairwise.perm.manova(Gene.dist, fraction_group, nperm = 999, p.method = "bonferroni")
Depth_ado
#L and H are not significantly different - in fact they match nearly perfectly



####Anosim and adonis date tests####

#anosim
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
date_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), date_group, distance = "bray")
date_ano$signif #0.012
date_ano$statistic #0.1105516
#Significant differences between group means - based on date

#Adonis
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
date_ado = adonis2(Gene.dist ~ date_group)
date_ado
#Significant differences between group means and distribution - based on date
#R2 = 0.21262 
#p = 0.001

#Pairwise PERMANOVA
date_group = get_variable(genecat_LH.ps@sam_data, "Sample_date")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
date.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, date_group, nperm = 999, p.method = "bonferroni")
date.stat
#All are significantly distinct from each other (p<0.05; 0.006-0.039)


####Anosim and adonis station_km tests####

#anosim
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
station_km_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), station_km_group, distance = "bray")
station_km_ano$signif #0.001
station_km_ano$statistic #0.3608903
#Significant differences between group means - based on station_km

#Adonis
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km_ado = adonis2(Gene.dist ~ station_km_group)
station_km_ado
#Significant differences between group means and distribution - based on station_km
#R2 = 0.29944
#p = 0.001

#Pairwise PERMANOVA
station_km_group = get_variable(genecat_LH.ps@sam_data, "station_km")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, station_km_group, nperm = 999, p.method = "bonferroni")
station_km.stat
#Most saline match each other (MG/ST/BB), while all samples match their preceding and following samples


####Anosim and adonis fraction tests####

#anosim
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
station_km_ano = anosim(phyloseq::distance(genecat_LH.ps, method = "bray"), station_km_group, distance = "bray")
station_km_ano$signif #0.897
station_km_ano$statistic #-0.02710472
#No significant differences between group means

#Adonis
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km_ado = adonis2(Gene.dist ~ station_km_group)
station_km_ado
#No significant differences between group means and distribution - based on fraction
#R2 = 0.00581  
#p = 0.958

#Pairwise PERMANOVA
station_km_group = get_variable(genecat_LH.ps@sam_data, "Sample_type")
Gene.dist = phyloseq::distance(genecat_LH.ps, method = "bray")
station_km.stat = RVAideMemoire::pairwise.perm.manova(Gene.dist, station_km_group, nperm = 999, p.method = "bonferroni")
station_km.stat
#No significant fraction difference


```
#Transcripts per gene copy
##Prepare data frame
```{r}

#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2) %>%
  distinct()

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = gsub("-", " ",amoAClean_CO2CH4_tbl_long$Sample_date)

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("May 21",
                                              "Jul 21",
                                              "Nov 21",
                                              "Feb 22",
                                              "May 22",
                                              "Jun 22",
                                              "Nov 22"))

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)


#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
CO2CH4_TG.df <- spread(amoAClean_CO2CH4_tbl_long[ , ! names(amoAClean_CO2CH4_tbl_long) %in% c("sampleid", "abundance", "occurrence", "X")], #Remove columns that will cause problems
                       data_type, #Column that will contain new column names
                       counts) %>% # Value that will fill new columns
  mutate_at(c("Metagenomes", "Transcriptomes"), ~replace(., is.na(.), 0)) #Replace NAs with 0 in the new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_tbl_long)
dim(CO2CH4_TG.df)




```

##Run Statistical Fraction test
```{r}

#Calculate Transcripts per gene copy
CO2CH4_TG.df$TranscriptsPerGenome = CO2CH4_TG.df$Transcriptomes / CO2CH4_TG.df$Metagenomes

#Add more info - remove NA's and Infinite numbers
CO2CH4_TG.df = CO2CH4_TG.df %>%
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.na(.), 0)) %>% # This was ususally the case when 0/0
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.infinite(.), "IncompleteMetagenome")) #Replace Infin with #/0, as Metagenome likely incomplete

#Convert from wide to long format
CO2CH4_TG.lng = gather(CO2CH4_TG.df, key = data_type, value = counts, Metagenomes:TranscriptsPerGenome)
dim(CO2CH4_TG.lng)

#Extract list of genes
uniquegenes.ls = unique(CO2CH4_TG.lng$KO)
uniquegenes.ls

#Make results table for metagenome vs metatranscriptomes
Results_MT.df = data.frame(Gene = "DELETEME",
                        Sample_type.TranscriptsPerGenome = 9999,
                        Sample_type.r = tmp.m$statistic,
                        Sample_type.p = 9999)
#Make results table for LH
Results_LH.df = data.frame(Gene = "DELETEME",
                           Suspended.TranscriptsPerGenome = 9999,
                           Sinking.TranscriptsPerGenome = 9999,
                           Sample_type.r = tmp.m$statistic,
                           Sample_type.p = 9999)

i="K22213"
#Get significant differences between metagenomes and metatranscriptomes and then Suspended and Sinking particles
for (i in uniquegenes.ls) {
  print(paste0("Working on ", i, ". Number ", which(uniquegenes.ls == i), " out of ", length(uniquegenes.ls)))
  
  #Subset to only one KO entry
  tmp.sbst = subset(CO2CH4_TG.df, KO == i & Sample_date!="Nov 21")
  
  #Remove samples with missing metagenome information
  tmp.sbst$TranscriptsPerGenome = as.numeric(tmp.sbst$TranscriptsPerGenome)
  
  #Add more info - remove NA's and Infinite numbers
  tmp.sbst = tmp.sbst %>%
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.na(.), 0))
  
  
  
  #Run significance test between metagenomes and metatranscriptome
  #Run mantel test
  test1 = vegdist(tmp.sbst$Metagenomes, method = "euclidean")
  test2 = vegdist(tmp.sbst$Transcriptomes, method = "euclidean")
  tmp.m = vegan::mantel(test1, test2)

  
  #Make temporary dataframe with results
  tmp.res = data.frame(Gene = paste0(i),
                       Sample_type.TranscriptsPerGenome = mean(tmp.sbst$TranscriptsPerGenome, na.rm = T),
                       Sample_type.r = tmp.m$statistic,
                       Sample_type.p = tmp.m$signif)
  
  
  Results_MT.df = rbind(Results_MT.df, tmp.res)
  
  
  
  
  
  #Run significance test between suspended and sinking particles
  
  #Separate into suspended and sinking particles
  tmp.l = subset(tmp.sbst, Sample_type == "Light_fraction")
  tmp.h = subset(tmp.sbst, Sample_type == "Heavy_fraction")
  #Subset for dataframes to match
  tmp.l = subset(tmp.l, BioSample!= "SAMEA110290340" & BioSample!= "SAMEA112714817")
  tmp.h = subset(tmp.h, BioSample!= "SAMEA112714807")
  
  #tmp.l[,c(1,7,11,14,25,29:32)]
  #tmp.h[,c(1,7,11,14,25,29:32)]
  print(paste0("Suspended particles have ", dim(tmp.l)[1], " samples, and sinking have ",  dim(tmp.h)[1]))
  
  #Run mantel test
  test1 = vegdist(tmp.l$TranscriptsPerGenome, method = "euclidean")
  test2 = vegdist(tmp.h$TranscriptsPerGenome, method = "euclidean")
  tmp.m = vegan::mantel(test1, test2)

  print("Test")
  
  #Make temporary dataframe with results
  tmp.res = data.frame(Gene = paste0(i),
                       Suspended.TranscriptsPerGenome = mean(tmp.l$TranscriptsPerGenome, na.rm = T),
                       Sinking.TranscriptsPerGenome = mean(tmp.h$TranscriptsPerGenome, na.rm = T),
                       Sample_type.r = tmp.m$statistic,
                       Sample_type.p = tmp.m$signif)
  
  
  Results_LH.df = rbind(Results_LH.df, tmp.res)
  
}
#Check results and clean up dataframe
Results_MT.df = Results_MT.df[-1,]
Results_MT.df
Results_LH.df = Results_LH.df[-1,]
Results_LH.df

#Subset to only significant gene hits
Results_SignMT.df = subset(Results_MT.df, Sample_type.p < 0.05)
dim(Results_SignMT.df)
Results_SignLH.df = subset(Results_LH.df, Sample_type.p < 0.05)
dim(Results_SignLH.df)


Top25_Fraction.df = Results_SignLH.df %>% filter(Sample_type.r > quantile(Sample_type.r, 0.75))

ggplot(test, aes(y = -log10(Sample_type.p), x = log2(Sample_type.r)))+
  geom_point()

Top25_Fraction.ls = Top25_Fraction.df$Gene


```

##Make Gene plot 
```{r}


####Create dataframe####
#Import gene list
CarbonGenelist.df = read.csv("CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2)

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

####Run test####

amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")
#Remove all but the relevant object to run anova models
#library(gdata)
#keep(amoAClean_CO2CH4_tbl_long)
#keep(amoAClean_CO2CH4_tbl_long, sure = TRUE, all = TRUE)

#CO2CH4_tbl_long %>% separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName"))

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("Mai 21",
                                              "Jul 21",
                                              "Feb 22",
                                              "Mai 22",
                                              "Jun 22",
                                              "Nov 22"))

#Adjust values for accuracy and log transform
#amoAClean_CO2CH4_tbl_long$counts = (amoAClean_CO2CH4_tbl_long$counts/1000)#+1
#CO2CH4_tbl_long$counts = CO2CH4_tbl_long$counts + 1

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)

#BugHunt.df = amoAClean_CO2CH4_tbl_long[c(1:200000),]

Complete_Metadata.df = read.csv("SAMEAID_SampleID.csv", sep = ";", header = T)[,c(1,10:15)]

amoAClean_CO2CH4_tbl_long = amoAClean_CO2CH4_tbl_long %>%
  select(-X) %>%
  left_join(Complete_Metadata.df, by = c("BioSample" = "AccessionNumber_TBDSven")) %>%
  mutate_at(c('counts'), as.numeric) %>%
  distinct()
dim(amoAClean_CO2CH4_tbl_long)



#Read in dataframe
SignGenes_Fraction.df = read.csv("GeneSignificant_Fraction.csv", header = T, sep = ";")

SignGenes_Fraction.df = subset(SignGenes_Fraction.df, Sample_type.eta2 > 0.3)
dim(SignGenes_Fraction.df)
# Plot preferences
amoAClean_CO2CH4Profiles.plt = ggplot(SignGenes_Fraction.df) + 
  geom_tile(aes(x=data_type, y=fct_reorder(.f = Gene, .x = Sample_type.eta2, .fun = mean, .na_rm = T, .desc = F), fill=Sample_type.eta2)) +
  #facet_wrap(Direction ~ data_type * Sample_date, ncol = 9)+
  scale_fill_gradient2(expression(paste("eta"^{2})), low= "magenta", mid = "white", high = "black", na.value = "red")+
  ggtitle("CO2/CH4")+
  ylab("Significantly changing \nCO2/CH4 Genes")+
  xlab("Sequencing type")+
  My_Theme+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# visualise
amoAClean_CO2CH4Profiles.plt 

#Save plots
#pdf("Figures/CO2CH4_Unknown Reversibility.pdf", width = 12, height = 12)
#CO2CH4Profiles.plt
#dev.off()

png("Figures/CO2CH4_SignfractionTop25.png", width = 10, height = 12, units = "in", res = 120)
amoAClean_CO2CH4Profiles.plt
dev.off()





 
```


#Physicochemical heatmap
```{r}
#Import as clean data frame
Physicochem.df = read.csv("PhysicochemicalParameters_mod.txt", encoding = "UTF-8", sep = "\t") %>%
  subset(Sample_date!="Nov 21") #Remove data taken from shore

#Clean up data  
Physicochem.df$station_km = as.numeric(Physicochem.df$Stromkilometer)
Physicochem.df$Stromkilometer = NULL
Physicochem.df$station_km = gsub(608.165, 608, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(613, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.88, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.884, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(633.022, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6334, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.32, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.323, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.955, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6513, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.41, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.414, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.546, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6654, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(691.997, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.010, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.01, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(694, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(711.515, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(712, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.975, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.98, 713, Physicochem.df$station_km)
sort(unique(Physicochem.df$station_km))

Physicochem.df = subset(Physicochem.df, station_km > 630)
colnames(Physicochem.df)
Physicochem.df = distinct(Physicochem.df)

Physicochem.df[c(8:37)] = sapply(Physicochem.df[c(8:37)], as.numeric)


#Make results dataframe
Results.df = data.frame(Var1 = as.character("DELETEME"),
                        Var2 = as.character("DELETEME"),
                        p.value = as.numeric(9999999),
                        rho.value = as.numeric(9999999))
#Run comparison tests
i=9
x=10
for (i in c(8:27,29:37)) {
  print(colnames(Physicochem.df)[i])
  
  for (x in c(8:27,29:37)) {
    
    #Remove problematic comparisons
    if (grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Total_DIN_uM") == T | 
        grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Ammonium_uM") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Nitrite_uM") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Nitrate_uM") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Phosphate_uM") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_mg.L") == T & grepl(colnames(Physicochem.df)[x], "Silicate_uM") == T |
      grepl(colnames(Physicochem.df)[i], "Total_DIN_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T | 
        grepl(colnames(Physicochem.df)[i], "Ammonium_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T |
      grepl(colnames(Physicochem.df)[i], "Nitrite_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T |
      grepl(colnames(Physicochem.df)[i], "Nitrate_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T |
      grepl(colnames(Physicochem.df)[i], "Phosphate_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_uM") == T & grepl(colnames(Physicochem.df)[x], "Silicate_mg.L") == T |
      
      
      grepl(colnames(Physicochem.df)[i], "Total_DIN_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T |
      grepl(colnames(Physicochem.df)[i], "Ammonium_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T |
      grepl(colnames(Physicochem.df)[i], "Nitrite_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T |
      grepl(colnames(Physicochem.df)[i], "Nitrate_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T |
      grepl(colnames(Physicochem.df)[i], "Phosphate_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T |
      grepl(colnames(Physicochem.df)[i], "Silicate_uM") == T & grepl(colnames(Physicochem.df)[x], "RespirationRate_O2ug.L.h") == T | 
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Total_DIN_uM") == T |
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Ammonium_uM") == T |
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Nitrite_uM") == T |
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Nitrate_uM") == T |
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Phosphate_uM") == T |
      grepl(colnames(Physicochem.df)[i], "RespirationRate_O2ug.L.h") == T & grepl(colnames(Physicochem.df)[x], "Silicate_uM") == T){
      
      print(paste0("Error with: i = ", colnames(Physicochem.df)[i] ," & x = ", colnames(Physicochem.df)[x]))
      
    } else {
  
    test.tmp = cor.test(as.numeric(unlist(Physicochem.df[i])), as.numeric(unlist(Physicochem.df[x])), method = "pearson") # we are looking for linear relationships, so we use a pearson instead of a spearman test (which also includes more complex monotonic relationships)
  
    tmp.df = data.frame(Var1 = colnames(Physicochem.df)[i],
                        Var2 = colnames(Physicochem.df)[x],
                        p.value = as.numeric(test.tmp$p.value),
                        rho.value = as.numeric(test.tmp$estimate))
    
    Results.df = rbind(Results.df, tmp.df)
  
}
  
  
}
}
#Remove initial row with DELETEME
Results.df = Results.df[-1,]

#Remove self comparisons
Results.df = Results.df %>% 
  as_tibble() %>% 
  mutate(duplicates = if_else(Var1 == Var2,
                              TRUE,
                              FALSE)) %>% 
  filter(duplicates == FALSE)

Results_sign.df = subset(Results.df, p.value < 0.05)
#Results_sign.df = Results.df

#Make p value heatmap
Physico_pvalue.plt = ggplot(Results_sign.df, aes(x = Var1, y = Var2, fill = p.value))+
  geom_tile()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_fill_gradient(low = "navy", high = "white")+
  xlab("Physicochemical parameters") + 
  ylab("Physicochemical parameters")
Physico_pvalue.plt

#Make rho value heatmap
Physico_rhovalue.plt =ggplot(Results.df, aes(x = Var1, y = Var2, fill = rho.value))+
  geom_tile()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_fill_gradient(low = "white", high = "navy")+
  xlab("Physicochemical parameters") + 
  ylab("Physicochemical parameters")
Physico_rhovalue.plt


#Test arrangement
ggarrange(Physico_pvalue.plt, Physico_rhovalue.plt)

png("Figures/Physicochem_heatmap_pearson.png", width = 16, height = 8, units = "in", res = 120)
ggarrange(Physico_pvalue.plt, Physico_rhovalue.plt)
dev.off()

#Cut off for very strong relationships is >0.8 according to Chan YH. Biostatistics 104: correlational analysis. Singapore Med J. 2003 Dec;44(12):614-9. PMID: 14770254.
#Subset to this
Results_sign_VS.df = subset(Results_sign.df[,-5], rho.value > 0.8 | rho.value < -0.8)

#Extract dataframe to identify representative physicochemical paremeters
write.csv(Results_sign_VS.df, "Physicochem_Pearson.csv")
```
##Identify significantly correlated parameters for later species, functional potential, and transcript per gene correlation analyses
```{r}
#Extract the individual physicochemical parameters that have to be used - for later 
Networked.ls = paste0(Results_sign_VS.df$Var1)

Networked_unique.ls = unique(Networked.ls)

#Extract the individual physicochemical parameters that have to be used
NonNetworked.ls = unique(paste0(Results.df$Var1))

NonNetworked_unique.ls = unique(NonNetworked.ls)


```
##Plot dCO2 and dCH4
```{r}


#Read in new PP dataframe as a clean data frame
Physicochem.df = read.csv("F:/Functional_R_analysis/PhysicochemicalParameters_mod.txt", encoding = "UTF-8", sep = "\t") %>%
  subset(Sample_date!="Nov 21") #Remove data taken from shore

#Clean up data  
Physicochem.df$station_km = as.numeric(Physicochem.df$Stromkilometer)
Physicochem.df$Stromkilometer = NULL
Physicochem.df$station_km = gsub(608.165, 608, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(613, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.88, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.884, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(633.022, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6334, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.32, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.323, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.955, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6513, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.41, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.414, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.546, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6654, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(691.997, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.010, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.01, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(694, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(711.515, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(712, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.975, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.98, 713, Physicochem.df$station_km)
sort(unique(Physicochem.df$station_km))

Physicochem.df = subset(Physicochem.df, station_km > 630)

#Rename rows for downstream applications
rownames(Physicochem.df) = Physicochem.df$Associatednumber

#Remove redundant ones identified with correlated pearson tests
PP_totest.ls = unique(colnames(Physicochem.df))
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Associatednumber"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "DNA_concentration_ng.uL"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Station"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "StationNumber"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample_type"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample_date"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "AccessionNumber_TBDSven"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "POC_mgperL"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "PTH_mgperL"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "TN_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Phosphate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Total_DIN_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Temperature_TBDHereon"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "Silicate_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Ammonium_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Nitrate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Nitrite_uM"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "Ammonium_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Silicate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "SRP_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "RespirationRate_O2ug.L.h"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "DIC_uM."]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "TotalDissolvedPhosphate_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sat_O2_Perc"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "dCH4_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "dCO2_nM"]  # Removes elements that are "b"

#Extract relevant columns
PP_totest.df = Physicochem.df  %>% select(all_of(PP_totest.ls)) %>%
  select(Sample_date, station_km, dCO2_uM, dCH4_nM, Salinity_PSU) %>%
  distinct()
dim(PP_totest.df)

#Convert relevant columns to numeric
PP_totest.df[,c(2:5)] = as.data.frame(sapply(PP_totest.df[,c(2:5)], as.numeric))

PP_totest.df$Sample_date = factor(PP_totest.df$Sample_date,
                                  levels = c("May 21",
                                             "Jul 21",
                                             "Feb 22",
                                             "May 22",
                                             "Jun 22",
                                             "Nov 22"))

CO2.plt = ggplot(PP_totest.df, aes(x = station_km, y = dCO2_uM, colour = Sample_date, group = Sample_date)) + 
  geom_point() + 
  geom_smooth() +
  scale_color_manual("Date", values = Date_colour_list)+
  scale_x_reverse()+
  My_Theme+
  xlab("Elbe km")+
  ylab(expression("Dissolved CO"[2]~"(ÂµM)"))

cor.test(PP_totest.df$dCO2_uM, PP_totest.df$station_km, method = "s")
cor.test(PP_totest.df$dCO2_uM, PP_totest.df$Salinity_PSU, method = "s")


CH4.plt = ggplot(PP_totest.df, aes(x = station_km, y = dCH4_nM, colour = Sample_date, group = Sample_date)) + 
  geom_point() + 
  geom_smooth() +
  scale_color_manual("Date", values = Date_colour_list)+
  scale_x_reverse()+
  My_Theme+
  xlab("Elbe km")+
  ylab(expression("Dissolved CH"[4]~"(nM)"))

cor.test(PP_totest.df$dCH4_nM, PP_totest.df$station_km, method = "s")
cor.test(PP_totest.df$dCH4_nM, PP_totest.df$Salinity_PSU, method = "s")


dC02dCH4.plt = ggarrange(CO2.plt, CH4.plt,
          common.legend = T,
          legend = "right")

png("F:/Functional_R_analysis/Figures/dCO2dCH4.png", units = "in", width = 8, height = 5, res = 120)
dC02dCH4.plt
dev.off()

```
#Physicochemical to microbiome aspects
##Set up physicochemical characteristics
```{r}

#Based on quick spearman correlation networks it's best to user PTH, phosphate, NH4, and TN for the networked physicochemical parameters - of course for not networked parameters the individual ones have to be chosen

#Based on quick spearman correlation networks it's best to user NO3 (Nitrate), PTC, DOC, DIC, pH, and NO2 (Nitrite) for the networked physicochemical parameters - of course for not networked parameters the individual ones have to be chosen
#Removing POC, PTH, TN, Phosphate, DIN, Temperature, Silicate, and NH4

#Read in new PP dataframe as a clean data frame
Physicochem.df = read.csv("F:/Functional_R_analysis/PhysicochemicalParameters_mod.txt", encoding = "UTF-8", sep = "\t") %>%
  subset(Sample_date!="Nov 21") #Remove data taken from shore

#Clean up data  
Physicochem.df$station_km = as.numeric(Physicochem.df$Stromkilometer)
Physicochem.df$Stromkilometer = NULL
Physicochem.df$station_km = gsub(608.165, 608, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(613, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.88, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(632.884, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(633.022, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6334, 633, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.32, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.323, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(651.955, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6513, 651, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.41, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.414, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(665.546, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(6654, 665, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(691.997, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.010, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(692.01, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(694, 692, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(711.515, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(712, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.975, 713, Physicochem.df$station_km)
Physicochem.df$station_km = gsub(714.98, 713, Physicochem.df$station_km)
sort(unique(Physicochem.df$station_km))

Physicochem.df = subset(Physicochem.df, station_km > 630)

#Rename rows for downstream applications
rownames(Physicochem.df) = Physicochem.df$Associatednumber

#Remove redundant ones identified with correlated pearson tests
PP_totest.ls = unique(colnames(Physicochem.df))
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "Associatednumber"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "DNA_concentration_ng.uL"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Station"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "StationNumber"]  # Removes elements that are "b"
#PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample_type"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sample_date"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "AccessionNumber_TBDSven"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "POC_mgperL"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "PTH_mgperL"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "TN_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Phosphate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Total_DIN_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Temperature_TBDHereon"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Silicate_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Ammonium_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Nitrate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Nitrite_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Ammonium_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Silicate_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "SRP_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "RespirationRate_O2ug.L.h"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "DIC_uM."]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "TotalDissolvedPhosphate_mg.L"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "Sat_O2_Perc"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "dCH4_uM"]  # Removes elements that are "b"
PP_totest.ls = PP_totest.ls[PP_totest.ls != "dCO2_nM"]  # Removes elements that are "b"

#Extract relevant columns
PP_totest.df = Physicochem.df  %>% select(all_of(PP_totest.ls))

#Make sure rownames have been preserved
rownames(PP_totest.df)
dim(PP_totest.df)

#Extract water specific parameters
PP_water.df = PP_totest.df %>%
  subset(Sample_type == "Free_living") %>%
  select(-SPM_mgperL, -TEP_um2perL, -CSP_um2perL, -PTC_mgperL , -PTN_mgperL, -DOC_uM.L, -DIC_uM.L, -Sample_type) %>%
  distinct()
#Extract particle specific parameters
PP_particle.df = PP_totest.df %>%
  subset(Sample_type!="Free_living") %>%
  select(Associatednumber, PTC_mgperL, PTN_mgperL, SPM_mgperL, TEP_um2perL, CSP_um2perL, -Sample_type) %>%
  distinct()

#remove NAs and convert columns to numeric
#PP_water.df = na.omit(PP_water.df)
PP_water.df = as.data.frame(sapply(PP_water.df[,c(1:13)], as.numeric))

#Apply row numbers from sample IDs
rownames(PP_water.df) = PP_water.df$Associatednumber
PP_water.df$Associatednumber = NULL
rownames(PP_particle.df) = PP_particle.df$Associatednumber
PP_particle.df$Associatednumber = NULL

#Flip so columns are samples for dissimilarity assessment
PP_water.df = t(PP_water.df)
PP_particle.df = t(PP_particle.df)

#Get sample names for microbiome comparison
Samples_water.ls = colnames(PP_water.df)
Samples_particle.ls = colnames(PP_particle.df)


```
##Prepare metabolic dataframes
###Metagenomes
```{r}

#Import gene list
CarbonGenelist.df = read.csv("F:/Functional_R_analysis/CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2) %>%
  distinct()

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = gsub("-", " ",amoAClean_CO2CH4_tbl_long$Sample_date)

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("May 21",
                                              "Jul 21",
                                              "Nov 21",
                                              "Feb 22",
                                              "May 22",
                                              "Jun 22",
                                              "Nov 22"))

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)

#Subset to split metagenomes and metatranscriptomes and remove superflous columns
amoAClean_CO2CH4_tbl_long_MG = subset(amoAClean_CO2CH4_tbl_long, data_type == "Metagenomes") %>%
  select(counts, KO, Associatednumber)
amoAClean_CO2CH4_tbl_long_MT = subset(amoAClean_CO2CH4_tbl_long, data_type == "Transcriptomes") %>%
  select(counts, KO, Associatednumber)

dim(amoAClean_CO2CH4_tbl_long_MT)
amoAClean_CO2CH4_tbl_long_MT = unique(amoAClean_CO2CH4_tbl_long_MT)
dim(amoAClean_CO2CH4_tbl_long_MT)
dim(amoAClean_CO2CH4_tbl_long_MG)
amoAClean_CO2CH4_tbl_long_MG = unique(amoAClean_CO2CH4_tbl_long_MG)
dim(amoAClean_CO2CH4_tbl_long_MG)



#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
amoAClean_CO2CH4_wide_MT <- data.frame(spread(amoAClean_CO2CH4_tbl_long_MT,
                       key = Associatednumber, #Column that will contain new column names
                       value = counts)) # Value that will fill new columns
amoAClean_CO2CH4_wide_MG <- data.frame(spread(amoAClean_CO2CH4_tbl_long_MG,
                       key = Associatednumber, #Column that will contain new column names
                       value = counts)) # Value that will fill new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_wide_MT)
View(amoAClean_CO2CH4_wide_MT)
dim(amoAClean_CO2CH4_wide_MG)
View(amoAClean_CO2CH4_wide_MG)

#Apply row names from gene KOs
rownames(amoAClean_CO2CH4_wide_MT) = amoAClean_CO2CH4_wide_MT$KO
amoAClean_CO2CH4_wide_MT$KO = NULL
rownames(amoAClean_CO2CH4_wide_MG) = amoAClean_CO2CH4_wide_MG$KO
amoAClean_CO2CH4_wide_MG$KO = NULL

#Remove x from column names
colnames(amoAClean_CO2CH4_wide_MT) = gsub("X", "", colnames(amoAClean_CO2CH4_wide_MT))
colnames(amoAClean_CO2CH4_wide_MG) = gsub("X", "", colnames(amoAClean_CO2CH4_wide_MG))

```
###Metatranscriptomes
```{r}

#Import gene list
CarbonGenelist.df = read.csv("F:/Functional_R_analysis/CarbonGeneList.csv", header = T, sep = ";")

#Subset for testing purposes
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Name == "methane monooxygenase")
#BugHunt_GeneList.df = subset(CarbonGenelist.df, Direction == "Utilising")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO)

#Correct abundance
amoAClean_geneabund_KO[c(2:264)] = amoAClean_geneabund_KO[c(2:264)] / 1000


amoAClean_CO2CH4_tbl_long <- amoAClean_geneabund_KO %>%
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  pivot_longer(!KO, names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, DESCRIPTION, KO, ) %>% # only keep relevant columns
  separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## 245 samples
  mutate(abundance = if_else(counts < 0.1, 0, 1)) %>% 
  group_by(KO) %>% 
  mutate(occurrence = sum(abundance)/nsamples) %>% 
  ungroup() %>% 
#  filter(occurrence > 0, occurrence < .3)  %>% 
#  select(taxa) %>%  
#  unique() ##790 Conditionally Rare Taxa,  mOTUs occurring above 0.1% in less than 30% of samples 
  filter(occurrence > 0.2) %>%
  distinct()

View(amoAClean_CO2CH4_tbl_long)
dim(amoAClean_CO2CH4_tbl_long)

#write.csv(amoAClean_CO2CH4_tbl_long, "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = gsub("-", " ",amoAClean_CO2CH4_tbl_long$Sample_date)

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("May 21",
                                              "Jul 21",
                                              "Nov 21",
                                              "Feb 22",
                                              "May 22",
                                              "Jun 22",
                                              "Nov 22"))

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAG", "Metagenomes", amoAClean_CO2CH4_tbl_long$data_type)
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)


#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
CO2CH4_TG.df <- spread(amoAClean_CO2CH4_tbl_long[ , ! names(amoAClean_CO2CH4_tbl_long) %in% c("sampleid", "abundance", "occurrence", "X")], #Remove columns that will cause problems
                       data_type, #Column that will contain new column names
                       counts) %>% # Value that will fill new columns
  mutate_at(c("Metagenomes", "Transcriptomes"), ~replace(., is.na(.), 0)) #Replace NAs with 0 in the new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_tbl_long)
dim(CO2CH4_TG.df)




#Calculate Transcripts per gene copy
CO2CH4_TG.df$TranscriptsPerGenome = CO2CH4_TG.df$Transcriptomes / CO2CH4_TG.df$Metagenomes

#Add more info - remove NA's and Infinite numbers
CO2CH4_TG.df = CO2CH4_TG.df %>%
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.na(.), 0)) %>% # This was ususally the case when 0/0
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.infinite(.), "9999999999")) #Replace Infin with #/0, as Metagenome likely incomplete

#Convert from wide to long format
CO2CH4_TG.lng = gather(CO2CH4_TG.df, key = data_type, value = counts, Metagenomes:TranscriptsPerGenome)
dim(CO2CH4_TG.lng)

#Remove data_types and columns that are no longer relevant 
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, data_type == "TranscriptsPerGenome") %>%
  select(counts, KO, Associatednumber)
dim(CO2CH4_TG.lng)


#Remove samples for which we have no metatranscriptomes
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, Associatednumber > 116)

#Remove samples for which sequencing failed
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, !(Associatednumber %in% c(127, 128, 140, 145, 146, 157, 169, 170,
                                                               119, 125, 126, 136, 137, 153, 154)))
#Make sure data is set up properly
CO2CH4_TG.lng$counts = as.numeric(as.character(CO2CH4_TG.lng$counts))
CO2CH4_TG.lng$Associatednumber = as.character(CO2CH4_TG.lng$Associatednumber)

#Long to wide
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
amoAClean_CO2CH4_wide_MT <- data.frame(spread(unique(CO2CH4_TG.lng),
                       key = Associatednumber, #Column that will contain new column names
                       value = counts)) # Value that will fill new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_wide_MT)
View(amoAClean_CO2CH4_wide_MT)

#Bug check
max(amoAClean_CO2CH4_wide_MT$X117) # should be ~917

#Apply row names from gene KOs
rownames(amoAClean_CO2CH4_wide_MT) = amoAClean_CO2CH4_wide_MT$KO
amoAClean_CO2CH4_wide_MT$KO = NULL


#Remove x from column names
colnames(amoAClean_CO2CH4_wide_MT) = gsub("X", "", colnames(amoAClean_CO2CH4_wide_MT))




```

##Match physicochemical and microbial samples and remove extras
```{r}
####Set up dataframes and dissimilarities####

#Extract genes from samples matching conditions
MG_water.df = amoAClean_CO2CH4_wide_MG[names(amoAClean_CO2CH4_wide_MG) %in% Samples_water.ls]
MG_particles.df = amoAClean_CO2CH4_wide_MG[names(amoAClean_CO2CH4_wide_MG) %in% Samples_particle.ls]
MT_water.df = amoAClean_CO2CH4_wide_MT[names(amoAClean_CO2CH4_wide_MT) %in% Samples_water.ls]
MT_particles.df = amoAClean_CO2CH4_wide_MT[names(amoAClean_CO2CH4_wide_MT) %in% Samples_particle.ls]
#Check to make sure it worked
dim(MG_water.df)
dim(MG_particles.df)
dim(MT_water.df)
dim(MT_particles.df)


#Extract only microbial matched samples of Physicochemicals for dissimilarity otherwise mantel doesn't work
PP_water_MG.df = subset(PP_water.df, select = colnames(MG_water.df))
PP_particles_MG.df = subset(PP_particle.df, select = colnames(MG_particles.df))
PP_water_MT.df = subset(PP_water.df, select = colnames(MT_water.df))
PP_particles_MT.df = subset(PP_particle.df, select = colnames(MT_particles.df))

dim(PP_water_MG.df)
dim(PP_particles_MG.df)
dim(PP_water_MT.df)
dim(PP_particles_MT.df)


#Run dissimilarity comparison between samples
##Microbial
MG_water.diss = vegan::vegdist(t(MG_water.df), method = "bray")
MG_particles.diss = vegan::vegdist(t(MG_particles.df), method = "bray")
MT_water.diss = vegan::vegdist(t(MT_water.df), method = "bray")
MT_particles.diss = vegan::vegdist(t(MT_particles.df), method = "bray")
#Pysicochemical
PP_water_MG.diss = vegan::vegdist(t(PP_water_MG.df), method = "bray", na.rm = T)
PP_particles_MG.diss = vegan::vegdist(t(PP_particles_MG.df), method = "bray", na.rm = T)
PP_water_MT.diss = vegan::vegdist(t(PP_water_MT.df), method = "bray", na.rm = T)
PP_particles_MT.diss = vegan::vegdist(t(PP_particles_MT.df), method = "bray", na.rm = T)



####Mantel tests general ####

#Whole microbial vs whole physicochemical
mantel(MG_water.diss, PP_water_MG.diss) #p = 0.001, R=0.2412
mantel(MG_particles.diss, PP_particles_MG.diss) #p = 0.001, R=0.1997
mantel(MT_water.diss, PP_water_MT.diss) #p = 0.504, R=-0.Ã1172
mantel(MT_particles.diss, PP_particles_MT.diss) #p = 0.003, R=0.2551


#Run loop for individual genes vs individual physicochemical parameters
#####MG water ####
Results_MG_water.df = data.frame("Gene_KO" = "DELETEME",
                     "PhysicochemicalParameter" = "DELETEME",
                     "pvalue" = 99999,
                     "Mantel_Rvalue" = 99999)
i=1
x=7
for (i in 1:length(rownames(MG_water.df))) {
  for (x in 1:length(rownames(PP_water_MG.df))) {
    
    #Extracting names
    M.name = rownames(MG_water.df[i,])
    PP.name = rownames(as.data.frame(PP_water_MG.df)[x,])
    
    #Make dissimilarity matrices
    M.tmp = vegdist(t(MG_water.df[i,]), method = "euclidean")
    PP.tmp = vegdist(PP_water_MG.df[x,], method = "euclidean", na.rm = T)
    
    #Run mantel test
    test.tmp = mantel(M.tmp, PP.tmp, na.rm = T)
    
    #Make df for results
    Results.tmp = data.frame("Gene_KO" = M.name,
                             "PhysicochemicalParameter" = PP.name,
                             "pvalue" = test.tmp$signif,
                             "Mantel_Rvalue" = test.tmp$statistic)
    
    #Combine results with previous dataframe
    Results_MG_water.df = rbind(Results_MG_water.df, Results.tmp)
    
    
  }
  
  print(paste0("Completed ", i, " out of ", length(rownames(MG_water.df)), " genes, which is ", round((i/length(rownames(MG_water.df)))*100, digits = 2), "%"))
  
}

#Check results and clean up dataframe
Results_MG_water.df = Results_MG_water.df[-1,]
Results_MG_water.df

#Subset to only significant gene hits
Results_MG_water.df = subset(Results_MG_water.df, pvalue < 0.05)
dim(Results_MG_water.df)

Top25_Fraction_MG_water.df = Results_MG_water.df %>% filter(Mantel_Rvalue > quantile(Mantel_Rvalue, 0.75))
dim(Top25_Fraction_MG_water.df)

length(unique(Top25_Fraction_MG_water.df$Gene_KO))
unique(Top25_Fraction_MG_water.df$PhysicochemicalParameter)


#####MG particles ####
Results_MG_particles.df = data.frame("Gene_KO" = "DELETEME",
                     "PhysicochemicalParameter" = "DELETEME",
                     "pvalue" = 99999,
                     "Mantel_Rvalue" = 99999)
i=1
x=1
for (i in 1:length(rownames(MG_particles.df))) {
  for (x in 1:length(rownames(PP_particles_MG.df))) {
    
    #Extracting names
    M.name = rownames(MG_particles.df[i,])
    PP.name = rownames(as.data.frame(PP_particles_MG.df)[x,])
    
    #Make dissimilarity matrices
    M.tmp = vegdist(t(MG_particles.df[i,]), method = "euclidean")
    PP.tmp = vegdist(PP_particles_MG.df[x,], method = "euclidean", na.rm = T)
    
    #Run mantel test
    test.tmp = mantel(M.tmp, PP.tmp, na.rm = T)
    
    #Make df for results
    Results.tmp = data.frame("Gene_KO" = M.name,
                             "PhysicochemicalParameter" = PP.name,
                             "pvalue" = test.tmp$signif,
                             "Mantel_Rvalue" = test.tmp$statistic)
    
    #Combine results with previous dataframe
    Results_MG_particles.df = rbind(Results_MG_particles.df, Results.tmp)
    
    
  }
  
  print(paste0("Completed ", i, " out of ", length(rownames(MG_particles.df)), " genes, which is ", round((i/length(rownames(MG_particles.df)))*100, digits = 2), "%"))
  
}

#Check results and clean up dataframe
Results_MG_particles.df = Results_MG_particles.df[-1,]
Results_MG_particles.df

#Subset to only significant gene hits
Results_MG_particles.df = subset(Results_MG_particles.df, pvalue < 0.05)
dim(Results_MG_particles.df)

Top25_Fraction_MG_particles.df = Results_MG_particles.df %>% filter(Mantel_Rvalue > quantile(Mantel_Rvalue, 0.75))
dim(Top25_Fraction_MG_particles.df)

length(unique(Top25_Fraction_MG_particles.df$Gene_KO))
unique(Top25_Fraction_MG_particles.df$PhysicochemicalParameter)



#####MT water ####
Results_MT_water.df = data.frame("Gene_KO" = "DELETEME",
                     "PhysicochemicalParameter" = "DELETEME",
                     "pvalue" = 99999,
                     "Mantel_Rvalue" = 99999)
i=1
x=7
for (i in 1:length(rownames(MT_water.df))) {
  for (x in 1:length(rownames(PP_water_MT.df))) {
    
    #Extracting names
    M.name = rownames(MT_water.df[i,])
    PP.name = rownames(as.data.frame(PP_water_MT.df)[x,])
    
    #Make dissimilarity matrices
    M.tmp = vegdist(t(MT_water.df[i,]), method = "euclidean")
    PP.tmp = vegdist(PP_water_MT.df[x,], method = "euclidean", na.rm = T)
    
    #Run mantel test
    test.tmp = mantel(M.tmp, PP.tmp, na.rm = T)
    
    #Make df for results
    Results.tmp = data.frame("Gene_KO" = M.name,
                             "PhysicochemicalParameter" = PP.name,
                             "pvalue" = test.tmp$signif,
                             "Mantel_Rvalue" = test.tmp$statistic)
    
    #Combine results with previous dataframe
    Results_MT_water.df = rbind(Results_MT_water.df, Results.tmp)
    
    
  }
  
  print(paste0("Completed ", i, " out of ", length(rownames(MT_water.df)), " genes, which is ", round((i/length(rownames(MT_water.df)))*100, digits = 2), "%"))
  
}

#Check results and clean up dataframe
Results_MT_water.df = Results_MT_water.df[-1,]
Results_MT_water.df

#Subset to only significant gene hits
Results_MT_water.df = subset(Results_MT_water.df, pvalue < 0.05)
dim(Results_MT_water.df)

Top25_Fraction_MT_water.df = Results_MT_water.df %>% filter(Mantel_Rvalue > quantile(Mantel_Rvalue, 0.75))
dim(Top25_Fraction_MT_water.df)

length(unique(Top25_Fraction_MT_water.df$Gene_KO))
unique(Top25_Fraction_MT_water.df$PhysicochemicalParameter)


#####MT particles ####
Results_MT_particles.df = data.frame("Gene_KO" = "DELETEME",
                     "PhysicochemicalParameter" = "DELETEME",
                     "pvalue" = 99999,
                     "Mantel_Rvalue" = 99999)
i=1
x=1
for (i in 1:length(rownames(MT_particles.df))) {
  for (x in 1:length(rownames(PP_particles_MT.df))) {
    
    #Extracting names
    M.name = rownames(MT_particles.df[i,])
    PP.name = rownames(as.data.frame(PP_particles_MT.df)[x,])
    
    #Make dissimilarity matrices
    M.tmp = vegdist(t(MT_particles.df[i,]), method = "euclidean")
    PP.tmp = vegdist(PP_particles_MT.df[x,], method = "euclidean", na.rm = T)
    
    #Run mantel test
    test.tmp = mantel(M.tmp, PP.tmp, na.rm = T)
    
    #Make df for results
    Results.tmp = data.frame("Gene_KO" = M.name,
                             "PhysicochemicalParameter" = PP.name,
                             "pvalue" = test.tmp$signif,
                             "Mantel_Rvalue" = test.tmp$statistic)
    
    #Combine results with previous dataframe
    Results_MT_particles.df = rbind(Results_MT_particles.df, Results.tmp)
    
    
  }
  
  print(paste0("Completed ", i, " out of ", length(rownames(MT_particles.df)), " genes, which is ", round((i/length(rownames(MT_particles.df)))*100, digits = 2), "%"))
  
}

#Check results and clean up dataframe
Results_MT_particles.df = Results_MT_particles.df[-1,]
Results_MT_particles.df

#Subset to only significant gene hits
Results_MT_particles.df = subset(Results_MT_particles.df, pvalue < 0.05)
dim(Results_MT_particles.df)

Top25_Fraction_MT_particles.df = Results_MT_particles.df %>% filter(Mantel_Rvalue > quantile(Mantel_Rvalue, 0.75))
dim(Top25_Fraction_MT_particles.df)

length(unique(Top25_Fraction_MT_particles.df$Gene_KO))
unique(Top25_Fraction_MT_particles.df$PhysicochemicalParameter)


#### Combine all results ####

Results_MG_water.df$data_type = "Metagenomes"
Results_MG_particles.df$data_type = "Metagenomes"
Results_MT_water.df$data_type = "Metatranscriptomes"
Results_MT_particles.df$data_type = "Metatranscriptomes"

Results_MG_water.df$environment = "Water"
Results_MG_particles.df$environment = "Particles"
Results_MT_water.df$environment = "Water"
Results_MT_particles.df$environment = "Particles"



Results.df = rbind(Results_MG_water.df, Results_MG_particles.df,
                   Results_MT_water.df, Results_MT_particles.df)

write.csv(Results.df, "Mantel_IndGenes_IndPP.csv")


####Analyse ####

Results.df = read.csv("Mantel_IndGenes_IndPP.csv")

Results.df = subset(Results.df, PhysicochemicalParameter!="station_km")
dim(Results.df)

Results_MG_water.df = subset(Results.df, data_type == "Metagenomes" & environment == "Water")
Results_MG_particles.df = subset(Results.df, data_type == "Metagenomes" & environment == "Particles")
Results_MT_water.df = subset(Results.df, data_type == "Metatranscriptomes" & environment == "Water")
Results_MT_particles.df = subset(Results.df, data_type == "Metatranscriptomes" & environment == "Particles")

ggplot(Results_MG_water.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=after_stat(count)), vjust=-1)

ggplot(Results_MG_particles.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=after_stat(count)), vjust=-1)

ggplot(Results_MT_water.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=after_stat(count)), vjust=-1)

ggplot(Results_MT_particles.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=after_stat(count)), vjust=-1)

#Water PP
Results_water.df = subset(Results.df, environment == "Water")

ggplot(Results_water.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=..count..), vjust=-1)

#Water PP
Results_particle.df = subset(Results.df, environment == "Particles")

ggplot(Results_particle.df, aes(x = PhysicochemicalParameter)) + 
  geom_bar() +
  geom_text(stat='count', aes(label=..count..), vjust=-1)
 

```

##Match to MAGs of origin

```{r Salinity MAG matching}
#### Metagenomes ####

#Extract relevant genes
Salinity_genes_MG.ls = sort(subset(Results_MG_water.df, PhysicochemicalParameter == "Salinity_PSU")[,1])
Salinity_genes_MG.ls = c("K00027", "K00029", "K00030", "K00122", "K00123", "K00126", "K00161", "K00162", "K00163", "K00164", "K00179", "K00180", "K00201", "K00228", "K00252", "K00281", "K00283", "K00365", "K00457", "K00471", "K00477", "K00595", "K00643", "K00647", "K00767", "K00788", "K01003", "K01428", "K01459", "K01484", "K01571", "K01575", "K01578", "K01579", "K01580", "K01584", "K01585", "K01586", "K01591", "K01593", "K01595", "K01596", "K01597", "K01607", "K01608", "K01609", "K01610", "K01613", "K01616", "K01653", "K01673", "K01674", "K01713", "K01725", "K01935", "K02548", "K02551", "K03119", "K03392", "K03518", "K03519", "K03520", "K03782", "K03928", "K04102", "K04517", "K04518", "K05886", "K05921", "K06016", "K06033", "K06034", "K08678", "K11381", "K12251", "K12253", "K13039", "K13485", "K13745", "K14446", "K15737", "K16066", "K16872", "K18076", "K18473", "K20036", "K20509", "K21615", "K21728", "K22024", "K22130", "K22616", "K23248")

#Import gene list
CarbonGenelist.df = read.csv("F:/Functional_R_analysis/CarbonGeneList.csv", header = T, sep = ";")

amoAMatches.ls = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% CarbonGenelist.df$KEGG_Orthology) %>% # only extract the CO2CH4 gene
  filter(str_detect(genus,"Nitros") & str_detect(KO, "^K10944$"))
dim(amoAMatches.ls)
sort(unique(amoAMatches.ls$KO))

amoAClean_geneabund_KO_sal = annot %>%
  rownames_to_column("gene_cluster") %>%
  left_join(gene_cluster_taxa, by=c("gene_cluster" = "gene_cluster")) %>% # add MAG information
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-many") %>% # add taxonomic information
  filter(KO %in% Salinity_genes_MG.ls) %>% # only extract the CO2CH4 gene
  anti_join(data.frame(gene = amoAMatches.ls$gene), by = "gene") %>% # Remove rows based on above taxa filtering
  select(gene_cluster, KO, genome) %>% # select only the columns titled gene_cluster and KO
  left_join(geneabund_2, by=c("gene_cluster" = "gene_cluster")) %>% # join these two columns to the gene abundance dataframe based on the gene_cluster column
  select(-length, -gene_cluster) %>% # remove the length and gene_cluster column
  group_by(genome, KO) %>% #aggregate the data by the KEGG ID
  summarise(across(everything(), ~ sum(., na.rm = TRUE))) # summarise the data using the sum to get abundance data - this still needs to be divided by 1000 for actual per genome values
dim(amoAClean_geneabund_KO_sal)
View(amoAClean_geneabund_KO_sal)

amoAClean_CO2CH4_tbl_long_sal <- amoAClean_geneabund_KO_sal %>%
  filter(KO %in% Salinity_genes_MG.ls) %>% # only extract the CO2CH4 gene
  pivot_longer(cols = -c(KO, genome), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% # reformat from a wide to long format, remove gene-cluster information, column names will be in the sampleid column, and values that would be NA are replaced with a 0.
  #left_join(annot, by = c("KO" = "KO"), relationship = "many-to-many") %>% # add gene information
  select(sampleid, counts, #DESCRIPTION,
         KO, genome, ) %>% # only keep relevant columns
  #separate(col = DESCRIPTION, sep = ";", into = c("SName", "LName")) %>% #separate description
  #left_join(CarbonGenelist.df, by = c("KO"="KEGG_Orthology")) %>%
  #select(!c(EC_Number, Pathway)) %>%
  left_join(metadata) %>% # add metadata
  filter(Sample_date != "Nov 21") %>% # remove samples from Nov 21
  filter(Station != 'BunthausSpitze') %>% #remove sample from BunthausSpitze
  filter(data_type == "METAG" & Sample_type == "Free_living") %>% # select only metatranscriptome samples
  ##Identify genes occurring at 0.1% abundance in at least 30% of samples
  ungroup() %>% 
  mutate(nsamples = n_distinct(sampleid)) %>%  ## XXX samples
  mutate(abundance = if_else(counts < 100, 0, 1)) %>% # 0.1% abundance
  group_by(KO, genome) %>% #group for occurrence
  mutate(occurrence = sum(abundance)/nsamples) %>% #calculate occurrence
  ungroup() %>%
  filter(occurrence > 0.1) %>% # filter for less than 1% occurrence
  distinct() %>%
  left_join(taxa, by = c("genome" = "GENOME"), relationship = "many-to-one")

#Correct abundance
amoAClean_CO2CH4_tbl_long_sal$counts = amoAClean_CO2CH4_tbl_long_sal$counts / 1000

#Remove phylogenetic level info
amoAClean_CO2CH4_tbl_long_sal$domain = gsub("d__", "", amoAClean_CO2CH4_tbl_long_sal$domain)
amoAClean_CO2CH4_tbl_long_sal$phylum = gsub("p__", "", amoAClean_CO2CH4_tbl_long_sal$phylum)
amoAClean_CO2CH4_tbl_long_sal$class = gsub("c__", "", amoAClean_CO2CH4_tbl_long_sal$class)
amoAClean_CO2CH4_tbl_long_sal$order = gsub("o__", "", amoAClean_CO2CH4_tbl_long_sal$order)
amoAClean_CO2CH4_tbl_long_sal$family = gsub("f__", "", amoAClean_CO2CH4_tbl_long_sal$family)
amoAClean_CO2CH4_tbl_long_sal$genus = gsub("g__", "", amoAClean_CO2CH4_tbl_long_sal$genus)
amoAClean_CO2CH4_tbl_long_sal$species = gsub("s__", "", amoAClean_CO2CH4_tbl_long_sal$species)

#Add full taxa info as a single column
amoAClean_CO2CH4_tbl_long_sal$fulltaxa = paste0(amoAClean_CO2CH4_tbl_long_sal$domain, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$phylum, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$class, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$order, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$family, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$genus, "__",
                                                amoAClean_CO2CH4_tbl_long_sal$species, "__")


#Check how many unique genera
unique(amoAClean_CO2CH4_tbl_long_sal$fulltaxa)

#Keep for records
write.csv(amoAClean_CO2CH4_tbl_long_sal, "F:/Functional_R_analysis/amoAClean_CO2CH4_OccurrenceFiltered_sal.csv")

#amoAClean_CO2CH4_tbl_long = read.csv( "amoAClean_CO2CH4_OccurrenceFiltered_V2.csv")

#Reorder factors
amoAClean_CO2CH4_tbl_long$Station = factor(amoAClean_CO2CH4_tbl_long$Station,
                                   levels = c("Medemgrund",
                                              "Brunsbuettel",
                                              "Schwarztonnensand",
                                              "Twielenfleth",
                                              "Muehlenberger Loch"))

amoAClean_CO2CH4_tbl_long$Sample_date = gsub("-", " ",amoAClean_CO2CH4_tbl_long$Sample_date)

amoAClean_CO2CH4_tbl_long$Sample_date = factor(amoAClean_CO2CH4_tbl_long$Sample_date,
                                   levels = c("May 21",
                                              "Jul 21",
                                              "Nov 21",
                                              "Feb 22",
                                              "May 22",
                                              "Jun 22",
                                              "Nov 22"))

#Make data_type names informative
amoAClean_CO2CH4_tbl_long$data_type = gsub("METAT", "Transcriptomes", amoAClean_CO2CH4_tbl_long$data_type)


#Long to wide, in preparation for transcripts per gene copy calculation
# The arguments to spread():
# - data: Data object
# - key: Name of column containing the new column names
# - value: Name of column containing values
CO2CH4_TG.df <- spread(amoAClean_CO2CH4_tbl_long[ , ! names(amoAClean_CO2CH4_tbl_long) %in% c("sampleid", "abundance", "occurrence", "X")], #Remove columns that will cause problems
                       data_type, #Column that will contain new column names
                       counts) %>% # Value that will fill new columns
  mutate_at(c("Metagenomes", "Transcriptomes"), ~replace(., is.na(.), 0)) #Replace NAs with 0 in the new columns
#Ensure that it worked
dim(amoAClean_CO2CH4_tbl_long)
dim(CO2CH4_TG.df)




#Calculate Transcripts per gene copy
CO2CH4_TG.df$TranscriptsPerGenome = CO2CH4_TG.df$Transcriptomes / CO2CH4_TG.df$Metagenomes

#Add more info - remove NA's and Infinite numbers
CO2CH4_TG.df = CO2CH4_TG.df %>%
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.na(.), 0)) %>% # This was ususally the case when 0/0
  mutate_at(c("TranscriptsPerGenome"), ~replace(., is.infinite(.), "9999999999")) #Replace Infin with #/0, as Metagenome likely incomplete

#Convert from wide to long format
CO2CH4_TG.lng = gather(CO2CH4_TG.df, key = data_type, value = counts, Metagenomes:TranscriptsPerGenome)
dim(CO2CH4_TG.lng)

#Remove data_types and columns that are no longer relevant 
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, data_type == "TranscriptsPerGenome") %>%
  select(counts, KO, Associatednumber)
dim(CO2CH4_TG.lng)


#Remove samples for which we have no metatranscriptomes
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, Associatednumber > 116)

#Remove samples for which sequencing failed
CO2CH4_TG.lng = subset(CO2CH4_TG.lng, !(Associatednumber %in% c(127, 128, 140, 145, 146, 157, 169, 170,
                                                               119, 125, 126, 136, 137, 153, 154)))
#Make sure data is set up properly
CO2CH4_TG.lng$counts = as.numeric(as.character(CO2CH4_TG.lng$counts))
CO2CH4_TG.lng$Associatednumber = as.character(CO2CH4_TG.lng$Associatednumber)



#STill need to calculate transcripts per gene


#### Metatranscriptomes ####


```


#Attic

```{r}

####Calculate variability####

#Extract ellipse coordinates from plot
pb = ggplot_build(NMDS_Gene.plt)
el = pb$data[[2]][c("colour", 
                    #"shape",
                    "x",
                    "y")]


i=1
ellipseAreaList = data.frame()
group_list = unique(el$colour)
#Separate into groups
for (i in 1:length(group_list)) {

  #Subset to separate into individual ellipse groups - if you used shape in addition to colour you'll want to change this as well
  Object = subset(el, 
                  colour == group_list[i])
  
  #Remove grouping column
  Object = Object[-1]
  
  # Center of ellipse
  ctr = MASS::cov.trob(Object)$center  
  
  # Calculate distance to center from each point on the ellipse
  dist2center <- sqrt(rowSums((t(t(Object)-ctr))^2))

  # Calculate area of ellipse from semi-major and semi-minor axes. 
  # These are, respectively, the largest and smallest values of dist2center. 
  value = pi*min(dist2center)*max(dist2center)
  
  #Store in the area list
  ellipseAreaList = rbind(ellipseAreaList, data.frame(group_list[i], value))
}

#Assign days from the colour scheme
ellipseAreaList$Date[grep("red", ellipseAreaList$group_list.i.)] = "Mai 21"
ellipseAreaList$Date[grep("forestgreen", ellipseAreaList$group_list.i.)] = "Jul 21"
ellipseAreaList$Date[grep("black", ellipseAreaList$group_list.i.)] = "Feb 22"
ellipseAreaList$Date[grep("green", ellipseAreaList$group_list.i.)] = "Mai 22"
ellipseAreaList$Date[grep("magenta", ellipseAreaList$group_list.i.)] = "Jun 22"
ellipseAreaList$Date[grep("navy", ellipseAreaList$group_list.i.)] = "Nov 22"

#Arrange for easier reading
ellipseAreaList = arrange(ellipseAreaList, desc(value))
ellipseAreaList

#Is the trend significant?
#cor.test(ellipseAreaList$value, ellipseAreaList$Date, method = "spearman")
kruskal.test(ellipseAreaList$value, ellipseAreaList$Date, p.adjust.method = "bonferroni") # this trend is not significant
```

